<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Unified Interface to spatial transcriptomics deconvolution tools ‚Ä¢ spacedeconv</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Unified Interface to spatial transcriptomics deconvolution tools">
<meta name="description" content="Collection of 31 deconvolution tools for spatial transcriptomics data.">
<meta property="og:description" content="Collection of 31 deconvolution tools for spatial transcriptomics data.">
<meta property="og:image" content="https://omnideconv.github.io/spacedeconv/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/spacedeconv.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/spacedeconv_clustering.html">Clustering</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_data.html">Data Input</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_example.html">Example</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_faq.html">FAQ</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_LR.html">Ligand-Receptor Analysis</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_utils.html">Workflow Utilities</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_visualization.html">Visualization</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv-decoupleR.html">DecoupleR integration</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="rocket-spacedeconv-">
<span class="emoji" data-emoji="rocket">üöÄ</span> spacedeconv 
<a class="anchor" aria-label="anchor" href="#rocket-spacedeconv-"></a>
</h1>
</div>
<p><a href="https://github.com/omnideconv/spacedeconv/actions/workflows/test.yml" class="external-link"><img src="https://github.com/omnideconv/spacedeconv/actions/workflows/test.yml/badge.svg" alt="R-CMD-check"></a> <a href="https://omnideconv.org/spacedeconv" class="external-link"><img src="https://github.com/omnideconv/spacedeconv/actions/workflows/pkgdown.yml/badge.svg" alt="docs"></a> <a href="https://github.com/omnideconv/spacedeconv/actions/workflows/test-coverage.yml" class="external-link"><img src="https://github.com/omnideconv/spacedeconv/actions/workflows/test-coverage.yml/badge.svg" alt="test-coverage"></a> <a href="https://codecov.io/gh/omnideconv/spacedeconv" class="external-link"><img src="https://codecov.io/gh/omnideconv/spacedeconv/graph/badge.svg?token=OX9ZHSEP9L" alt="codecov"></a></p>
<p>spacedeconv is a unified interface to 31 deconvolution tools with focus on spatial transcriptomics datasets. The package is able to directly estimate celltype proportions of immune cells and can deconvolute any celltype if an annotated single-cell reference dataset is available.</p>
<div class="section level2">
<h2 id="arrow_down-installation">
<span class="emoji" data-emoji="arrow_down">‚¨áÔ∏è</span> Installation<a class="anchor" aria-label="anchor" href="#arrow_down-installation"></a>
</h2>
<p><code>spacedeconv</code> is available from GitHub only. We recommend installing trough the pak package manager:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install the pak package manager</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"pak"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># recommended installation, deconvolution tools are installed on-demand</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/spacedeconv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># full installation including all deconvolution tools</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/spacedeconv"</span>, dependencies<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sparkles-features">
<span class="emoji" data-emoji="sparkles">‚ú®</span> Features<a class="anchor" aria-label="anchor" href="#sparkles-features"></a>
</h2>
<ul>
<li>unified access to 31 deconvolution tools</li>
<li>direct deconvolution of immune cells</li>
<li>compute custom reference signatures to deconvolute any celltype</li>
<li>flexible visualization functions</li>
<li>resource optimization</li>
<li>Pathway and Transcription Factor analysis (decoupleR integration)</li>
<li>Ligand-Receptor analysis</li>
<li>easy integration into spatial transcriptomics workflows</li>
</ul>
</div>
<div class="section level2">
<h2 id="floppy_disk-data-requirements">
<span class="emoji" data-emoji="floppy_disk">üíæ</span> Data requirements<a class="anchor" aria-label="anchor" href="#floppy_disk-data-requirements"></a>
</h2>
<p>Spatial transcriptomics data: <em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html" class="external-link">SpatialExperiment</a></em></p>
<p>Single-cell data with cell-type annotation: <em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html" class="external-link">SingleCellExperiment</a></em> (recommended), <em><a href="https://anndata.dynverse.org/" class="external-link">anndata</a></em> or <em><a href="https://satijalab.org/seurat/" class="external-link">Seurat</a></em></p>
</div>
<div class="section level2">
<h2 id="technologist-usage">
<span class="emoji" data-emoji="technologist">üßë‚Äçüíª</span> Usage<a class="anchor" aria-label="anchor" href="#technologist-usage"></a>
</h2>
<p>The main workflow consists of:</p>
<ol style="list-style-type: decimal">
<li>Reference signature computation using annotated single-cell data</li>
<li>Deconvolution</li>
<li>Visualization</li>
</ol>
<div class="section level3">
<h3 id="id_1-build-a-signature-matrix">1. Build a Signature Matrix<a class="anchor" aria-label="anchor" href="#id_1-build-a-signature-matrix"></a>
</h3>
<p>Build a cell type specific signature matrix from annotated single-cell reference data.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  <span class="va">single_cell_object</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spotlight"</span>,</span>
<span>  assay_sc<span class="op">=</span><span class="st">"cpm"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_2-deconvolution">2. Deconvolution<a class="anchor" aria-label="anchor" href="#id_2-deconvolution"></a>
</h3>
<p>While some methods are able to directly estimate immune cell abundances other tools require a custom reference signature computed in step 1).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  <span class="va">spatial_object</span>,</span>
<span>  <span class="va">signature</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spotlight"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_3-visualization">3. Visualization<a class="anchor" aria-label="anchor" href="#id_3-visualization"></a>
</h3>
<p>spacedeconv includes highly-flexible visualization functions. A full explanation of all visualization options can be found in the visualization <a href="articles/spacedeconv_visualization.html">vignette</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">spe</span>, cell_type<span class="op">=</span><span class="st">"spotlight_B.cells"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="bulb-additional-requirements">
<span class="emoji" data-emoji="bulb">üí°</span> Additional Requirements<a class="anchor" aria-label="anchor" href="#bulb-additional-requirements"></a>
</h2>
<p>Most methods do not require additional software/tokens, but there are a few exceptions:</p>
<ul>
<li>A working version of Docker is required for CIBERSORTx</li>
<li>A token for CIBERSORTx is required from this website: <a href="https://cibersortx.stanford.edu/" class="external-link uri">https://cibersortx.stanford.edu/</a>
</li>
<li>The CIBERSORT source code is required for BSeq-sc (see tutorial in ?omnideconv::bseqsc_config)</li>
<li>SpatialExperiment requires <code>magick</code> to be installed: <code>sudo apt-get install libmagick++-dev</code>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="available-methods-licenses-citations">Available methods, Licenses, Citations<a class="anchor" aria-label="anchor" href="#available-methods-licenses-citations"></a>
</h2>
<p>Note that, while <em>spacedeconv</em> itself is free (<a href="https://github.com/omnideconv/omnideconv/blob/main/LICENSE" class="external-link">GPL 3.0</a>), you may need to obtain a license to use the individual methods. See the table below for more information. If you use this package in your work, please cite both our package and the method(s) you are using.</p>
<blockquote>
<p>Constantin Zackl, Maria Zopoglou, Reto Stauffer, Markus Ausserhofer, Marieke E. Ijsselsteijn, Gregor Sturm, Noel Filipe da Cunha Carvalho de Miranda, Francesca Finotello. spacedeconv: deconvolution of tissue architecture from spatial transcriptomics, PREPRINT available at Research Square <a href="https://doi.org/10.21203/rs.3.rs-5102166/v1" class="external-link uri">https://doi.org/10.21203/rs.3.rs-5102166/v1</a></p>
</blockquote>
<table class="table">
<colgroup>
<col width="33%">
<col width="13%">
<col width="53%">
</colgroup>
<thead><tr class="header">
<th>First-gen (immunedeconv)</th>
<th>First-gen mouse (immunedeconv)</th>
<th>Second-gen (omnideconv + spatial Methods)</th>
</tr></thead>
<tbody><tr class="odd">
<td></td>
<td></td>
<td></td>
</tr></tbody>
</table>
</div>
</div>
<div class="section level1">
<h1 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<table class="table">
<colgroup>
<col width="14%">
<col width="4%">
<col width="18%">
<col width="61%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th align="center">signature</th>
<th align="center">licence</th>
<th>citation</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://github.com/omnideconv/omnideconv" class="external-link">omnideconv</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/omnideconv/omnideconv/blob/main/LICENSE" class="external-link">GPL-3</a></td>
<td>Citation will follow</td>
</tr>
<tr class="even">
<td><a href="https://github.com/omnideconv/immunedeconv" class="external-link">immunedeconv</a></td>
<td align="center"><span class="emoji" data-emoji="x">‚ùå</span></td>
<td align="center"><a href="https://github.com/omnideconv/immunedeconv/blob/master/LICENSE" class="external-link">BSD</a></td>
<td>Sturm, G., Finotello, F., Petitprez, F., Zhang, J. D., Baumbach, J., Fridman, W. H., ‚Ä¶, List, M., Aneichyk, T. (2019). Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology. Bioinformatics, 35(14), i436-i445.</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/RubD/Giotto/" class="external-link">spatialDWLS</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/RubD/Giotto/blob/master/LICENSE" class="external-link">MIT</a></td>
<td>Dong R, Yuan GC. SpatialDWLS: accurate deconvolution of spatial transcriptomic data. Genome Biol. 2021 May 10;22(1):145. doi: 10.1186/s13059-021-02362-7</td>
</tr>
<tr class="even">
<td><a href="https://github.com/BayraktarLab/cell2location" class="external-link">cell2location</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/BayraktarLab/cell2location/blob/master/LICENSE" class="external-link">Apache-2.0</a></td>
<td>Kleshchevnikov, V., Shmatko, A., Dann, E. et al.¬†Cell2location maps fine-grained cell types in spatial transcriptomics. Nat Biotechnol (2022). <a href="https://doi.org/10.1038/s41587-021-01139-4" class="external-link uri">https://doi.org/10.1038/s41587-021-01139-4</a>
</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/MarcElosua/SPOTlight" class="external-link">SPOTlight</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/MarcElosua/SPOTlight/blob/main/LICENSE.md" class="external-link">GPL 3.0</a></td>
<td>Elosua-Bayes M, Nieto P, Mereu E, Gut I, Heyn H (2021): <em>SPOTlight: seeded NMF regression to deconvolute spatial transcriptomics spots with single-cell transcriptomes</em>. <strong>Nucleic Acids Res</strong> 49(9):e50. <a href="doi:10.1093/nar/gkab043" class="uri">doi:10.1093/nar/gkab043</a>.</td>
</tr>
<tr class="even">
<td><a href="https://github.com/dmcable/spacexr" class="external-link">RCTD</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/dmcable/spacexr/blob/master/LICENSE" class="external-link">GPL 3.0</a></td>
<td>Cable, D.M., Murray, E., Zou, L.S. et al.¬†Robust decomposition of cell type mixtures in spatial transcriptomics. Nat Biotechnol 40, 517‚Äì526 (2022). <a href="https://doi.org/10.1038/s41587-021-00830-w" class="external-link uri">https://doi.org/10.1038/s41587-021-00830-w</a>
</td>
</tr>
<tr class="odd">
<td><a href="https://github.com/YingMa0107/CARD" class="external-link">CARD</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/YingMa0107/CARD/blob/master/LICENSE.md" class="external-link">GPL-3.0</a></td>
<td>Ying Ma, and Xiang Zhou (2022). Spatially informed cell type deconvolution for spatial transcriptomics. <a href="https://doi.org/10.1038/s41587-022-01273-7" class="external-link uri">https://doi.org/10.1038/s41587-022-01273-7</a>
</td>
</tr>
</tbody>
</table>
</div>

  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing spacedeconv</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Constantin Zackl <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1991-6943" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
