<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Unified Interface to spatial transcriptomics deconvolution tools ‚Ä¢ spacedeconv</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="favicon-96x96.png">
<link rel="icon" type="‚Äùimage/svg+xml‚Äù" href="favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png">
<link rel="icon" sizes="any" href="favicon.ico">
<link rel="manifest" href="site.webmanifest">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Unified Interface to spatial transcriptomics deconvolution tools">
<meta name="description" content="Collection of 31 deconvolution tools for spatial transcriptomics data.">
<meta property="og:description" content="Collection of 31 deconvolution tools for spatial transcriptomics data.">
<meta property="og:image" content="https://omnideconv.github.io/spacedeconv/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/spacedeconv_clustering.html">Clustering</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_data.html">Data Input</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_decoupleR.html">DecoupleR integration</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_example.html">Example</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_faq.html">FAQ</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_LR.html">Ligand-Receptor Analysis</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_utils.html">Workflow Utilities</a></li>
    <li><a class="dropdown-item" href="articles/spacedeconv_visualization.html">Visualization</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="rocket-spacedeconv-">
<span class="emoji" data-emoji="rocket">üöÄ</span> spacedeconv 
<a class="anchor" aria-label="anchor" href="#rocket-spacedeconv-"></a>
</h1>
</div>
<p><a href="https://www.gnu.org/licenses/gpl-3.0" class="external-link"><img src="https://img.shields.io/badge/license-GPL--3.0-blue.svg" alt="License: GPL-3.0"></a> <img src="https://img.shields.io/badge/R-4.3.3-276DC3?logo=r&amp;logoColor=white" alt="R"><img src="https://img.shields.io/badge/Python-3.10-3776AB?logo=python&amp;logoColor=white" alt="Python"><img src="https://img.shields.io/badge/platform-Linux%2064--bit-lightgrey" alt="Platform"></p>
<p><em>spacedeconv</em> is a unified interface to first- and second-generation deconvolution tools with focus on spatial transcriptomics datasets. The package is able to directly estimate celltype proportions of immune cells and can deconvolute any celltype if an annotated single-cell reference dataset is available.</p>
<div class="section level2">
<h2 id="arrow_down-installation">
<span class="emoji" data-emoji="arrow_down">‚¨áÔ∏è</span> Installation<a class="anchor" aria-label="anchor" href="#arrow_down-installation"></a>
</h2>
<p><strong>Note:</strong> The current <em>spacedeconv</em> installation is only available for the <strong>linux-64</strong> platform.</p>
<p>Since many different packages need to be included, we highly recommend to install <em>spacedeconv</em> in a new Conda environment with the following commands.</p>
<p>First, a tool for fast dependency resolution is needed, therefore we recommend installing mamba if not already available:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>conda install <span class="sc">-</span>c conda<span class="sc">-</span>forge mamba</span></code></pre></div>
<p>Download the environment.yml file of this github repo:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>wget https<span class="sc">:</span><span class="er">//</span>raw.githubusercontent.com<span class="sc">/</span>omnideconv<span class="sc">/</span>spacedeconv<span class="sc">/</span>main<span class="sc">/</span>environment.yml <span class="sc">-</span>O environment.yml</span></code></pre></div>
<p>Create a new environment called ‚Äúr-omnideconv‚Äù via mamba with the environment.yml file and flexible channel priority:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>CONDA_CHANNEL_PRIORITY<span class="ot">=</span>flexible mamba env create <span class="sc">-</span>f environment.yml</span></code></pre></div>
<p>Start R inside the r-omnideconv conda environment:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>conda activate r<span class="sc">-</span>omnideconv</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>R</span></code></pre></div>
<p>Install the missing packages that are not available via conda as well as <em>spacedeconv</em>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"drieslab/Giotto@v3.3.2"</span>, upgrade <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"YingMa0107/CARD"</span>, ref <span class="op">=</span> <span class="st">"2d64b91abb5cdd0c7f576b1c5d4727c84e7c93a0"</span>, upgrade <span class="op">=</span> <span class="st">"never"</span><span class="op">)</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/spacedeconv"</span>, dependencies <span class="op">=</span> <span class="cn">FALSE</span>, upgrade <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sparkles-features">
<span class="emoji" data-emoji="sparkles">‚ú®</span> Features<a class="anchor" aria-label="anchor" href="#sparkles-features"></a>
</h2>
<ul>
<li>unified access to first- and second-generation deconvolution tools</li>
<li>direct deconvolution of immune cells</li>
<li>compute custom reference signatures to deconvolute any celltype</li>
<li>flexible visualization functions</li>
<li>resource optimization</li>
<li>Pathway and Transcription Factor analysis (decoupleR integration)</li>
<li>Ligand-Receptor analysis</li>
<li>easy integration into spatial transcriptomics workflows</li>
</ul>
</div>
<div class="section level2">
<h2 id="floppy_disk-data-requirements">
<span class="emoji" data-emoji="floppy_disk">üíæ</span> Data requirements<a class="anchor" aria-label="anchor" href="#floppy_disk-data-requirements"></a>
</h2>
<p>Spatial transcriptomics data: <em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html" class="external-link">SpatialExperiment</a></em></p>
<p>Single-cell data with cell-type annotation: <em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html" class="external-link">SingleCellExperiment</a></em> (recommended) or <em><a href="https://satijalab.org/seurat/" class="external-link">Seurat</a></em></p>
</div>
<div class="section level2">
<h2 id="technologist-usage">
<span class="emoji" data-emoji="technologist">üßë‚Äçüíª</span> Usage<a class="anchor" aria-label="anchor" href="#technologist-usage"></a>
</h2>
<p>The main workflow consists of the following steps:</p>
<div class="section level3">
<h3 id="id_1-load-datasets">1. Load datasets<a class="anchor" aria-label="anchor" href="#id_1-load-datasets"></a>
</h3>
<p>To explore the package, start by loading some of the built-in example datasets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://omnideconv.github.io/spacedeconv">spacedeconv</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># data("single_cell_data_1")</span></span>
<span><span class="co"># data("single_cell_data_2")</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_3"</span><span class="op">)</span></span>
<span><span class="co"># data("single_cell_data_4")</span></span>
<span></span>
<span><span class="co"># data("spatial_data_1")</span></span>
<span><span class="co"># data("spatial_data_2")</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_3"</span><span class="op">)</span></span>
<span><span class="co"># data("spatial_data_4")</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_2-preprocessing">2. Preprocessing<a class="anchor" aria-label="anchor" href="#id_2-preprocessing"></a>
</h3>
<p>Depending on the use case, certain preprocessing steps might be necessary.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">spatial_data_3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spatial_data_3</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_3-build-a-signature-matrix">3. Build a Signature Matrix<a class="anchor" aria-label="anchor" href="#id_3-build-a-signature-matrix"></a>
</h3>
<p>Build a cell type specific signature matrix from annotated single-cell reference data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spatialdwls"</span>, verbose <span class="op">=</span> <span class="cn">T</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_4-deconvolution">4. Deconvolution<a class="anchor" aria-label="anchor" href="#id_4-deconvolution"></a>
</h3>
<p>While some methods are able to directly estimate immune cell abundances other tools require a custom reference signature.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deconv</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  spatial_obj <span class="op">=</span> <span class="va">spatial_data_3</span>,</span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spatialdwls"</span>,</span>
<span>  signature <span class="op">=</span> <span class="va">signature</span>,</span>
<span>  assay_sp <span class="op">=</span> <span class="st">"cpm"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_5-visualization">5. Visualization<a class="anchor" aria-label="anchor" href="#id_5-visualization"></a>
</h3>
<p><em>spacedeconv</em> includes highly-flexible visualization functions.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_spatial.html">plot_spatial</a></span><span class="op">(</span></span>
<span>  spe <span class="op">=</span> <span class="va">deconv</span>,</span>
<span>  result <span class="op">=</span> <span class="st">"spatialdwls_B.cells"</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"B cells"</span>,</span>
<span>  density<span class="op">=</span><span class="cn">F</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="available-methods-licenses-citations">Available methods, Licenses, Citations<a class="anchor" aria-label="anchor" href="#available-methods-licenses-citations"></a>
</h2>
<p>Note that, while <em>spacedeconv</em> itself is free (<a href="https://github.com/omnideconv/omnideconv/blob/main/LICENSE" class="external-link">GPL 3.0</a>), you may need to obtain a license to use the individual methods. See the table below for more information. If you use this package in your work, please cite both our package and the method(s) you are using.</p>
<blockquote>
<p>Constantin Zackl, Maria Zopoglou, Reto Stauffer, Markus Ausserhofer, Marieke E. Ijsselsteijn, Gregor Sturm, Noel Filipe da Cunha Carvalho de Miranda, Francesca Finotello. spacedeconv: deconvolution of tissue architecture from spatial transcriptomics, PREPRINT available at Research Square <a href="https://doi.org/10.21203/rs.3.rs-5102166/v1" class="external-link uri">https://doi.org/10.21203/rs.3.rs-5102166/v1</a>.</p>
</blockquote>
<table class="table">
<colgroup>
<col width="15%">
<col width="5%">
<col width="20%">
<col width="58%">
</colgroup>
<thead><tr>
<th>Method</th>
<th align="center">Usable with signature()</th>
<th align="center">Licence</th>
<th>Citation</th>
</tr></thead>
<tbody>
<tr>
<td><a href="https://github.com/omnideconv/immunedeconv" class="external-link">immunedeconv</a></td>
<td align="center"><span class="emoji" data-emoji="x">‚ùå</span></td>
<td align="center"><a href="https://github.com/omnideconv/immunedeconv/blob/master/LICENSE" class="external-link">BSD</a></td>
<td>Sturm, G. et al.¬†Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology. Bioinformatics, 35(14), i436-i445 (2019). <a href="https://doi.org/10.1093/bioinformatics/btz363" class="external-link uri">https://doi.org/10.1093/bioinformatics/btz363</a>
</td>
</tr>
<tr>
<td><a href="https://github.com/RubD/Giotto/" class="external-link">spatialDWLS</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/RubD/Giotto/blob/master/LICENSE" class="external-link">MIT</a></td>
<td>Dong R., Yuan G.C. SpatialDWLS: accurate deconvolution of spatial transcriptomic data. Genome Biology 22, 145 (2021). <a href="https://doi.org/10.1186/s13059-021-02362-7" class="external-link uri">https://doi.org/10.1186/s13059-021-02362-7</a>
</td>
</tr>
<tr>
<td><a href="https://github.com/BayraktarLab/cell2location" class="external-link">cell2location</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/BayraktarLab/cell2location/blob/master/LICENSE" class="external-link">Apache-2.0</a></td>
<td>Kleshchevnikov V. et al.¬†Cell2location maps fine-grained cell types in spatial transcriptomics. Nature Biotechnology 40, 661‚Äì671 (2022). <a href="https://doi.org/10.1038/s41587-021-01139-4" class="external-link uri">https://doi.org/10.1038/s41587-021-01139-4</a>
</td>
</tr>
<tr>
<td><a href="https://github.com/MarcElosua/SPOTlight" class="external-link">SPOTlight</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/MarcElosua/SPOTlight/blob/main/LICENSE.md" class="external-link">GPL 3.0</a></td>
<td>Elosua-Bayes M. et al.¬†SPOTlight: seeded NMF regression to deconvolute spatial transcriptomics spots with single-cell transcriptomes. Nucleic Acids Research 49(9):e50 (2021). <a href="https://doi.org/10.1093/nar/gkab043" class="external-link uri">https://doi.org/10.1093/nar/gkab043</a>.</td>
</tr>
<tr>
<td><a href="https://github.com/dmcable/spacexr" class="external-link">RCTD</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/dmcable/spacexr/blob/master/LICENSE" class="external-link">GPL 3.0</a></td>
<td>Cable D.M. et al.¬†Robust decomposition of cell type mixtures in spatial transcriptomics. Nature Biotechnology 40, 517‚Äì526 (2022). <a href="https://doi.org/10.1038/s41587-021-00830-w" class="external-link uri">https://doi.org/10.1038/s41587-021-00830-w</a>
</td>
</tr>
<tr>
<td><a href="https://github.com/YingMa0107/CARD" class="external-link">CARD</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/YingMa0107/CARD/blob/master/LICENSE.md" class="external-link">GPL-3.0</a></td>
<td>Ma Y., Zhou X. Spatially informed cell type deconvolution for spatial transcriptomics. Nature Biotechnology 40, 1349‚Äì1359 (2022). <a href="https://doi.org/10.1038/s41587-022-01273-7" class="external-link uri">https://doi.org/10.1038/s41587-022-01273-7</a>
</td>
</tr>
<tr>
<td><a href="https://github.com/saezlab/dot" class="external-link">DOT</a></td>
<td align="center"><span class="emoji" data-emoji="heavy_check_mark">‚úîÔ∏è</span></td>
<td align="center"><a href="https://github.com/saezlab/DOT/blob/main/LICENSE.md" class="external-link">GPL-3.0</a></td>
<td>Rahimi A. et al.¬†DOT: a flexible multi-objective optimization framework for transferring features across single-cell and spatial omics. Nature Communications 15, 4994 (2024). <a href="https://www.nature.com/articles/s41467-024-48868-z" class="external-link uri">https://www.nature.com/articles/s41467-024-48868-z</a>
</td>
</tr>
</tbody>
</table>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing spacedeconv</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Constantin Zackl <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1991-6943" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
