<!doctype html>
<!-- Generated by pkgdown: do not edit by hand -->
<html lang="en">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
    />
    <title>Workflow Utilities • spacedeconv</title>
    <!-- favicons -->
    <link
      rel="icon"
      type="image/png"
      sizes="96x96"
      href="../favicon-96x96.png"
    />
    <link rel="icon" type="”image/svg+xml”" href="../favicon.svg" />
    <link
      rel="apple-touch-icon"
      sizes="180x180"
      href="../apple-touch-icon.png"
    />
    <link rel="icon" sizes="any" href="../favicon.ico" />
    <link rel="manifest" href="../site.webmanifest" />
    <script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
    />
    <link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet" />
    <script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script>
    <link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
    <link
      href="../deps/font-awesome-6.5.2/css/v4-shims.min.css"
      rel="stylesheet"
    />
    <script src="../deps/headroom-0.11.0/headroom.min.js"></script>
    <script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script>
    <script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script>
    <script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script>
    <script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script>
    <script src="../deps/search-1.0.0/fuse.min.js"></script>
    <script src="../deps/search-1.0.0/mark.min.js"></script>
    <!-- pkgdown -->
    <script src="../pkgdown.js"></script>
    <meta property="og:title" content="Workflow Utilities" />
  </head>
  <body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>

    <nav
      class="navbar navbar-expand-lg fixed-top bg-light"
      data-bs-theme="light"
      aria-label="Site navigation"
    >
      <div class="container">
        <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

        <small
          class="nav-text text-muted me-auto"
          data-bs-toggle="tooltip"
          data-bs-placement="bottom"
          title=""
          >0.1.1</small
        >

        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbar"
          aria-controls="navbar"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>

        <div id="navbar" class="collapse navbar-collapse ms-3">
          <ul class="navbar-nav me-auto">
            <li class="nav-item">
              <a class="nav-link" href="../articles/spacedeconv.html"
                >Get started</a
              >
            </li>
            <li class="nav-item">
              <a class="nav-link" href="../reference/index.html">Reference</a>
            </li>
            <li class="active nav-item dropdown">
              <button
                class="nav-link dropdown-toggle"
                type="button"
                id="dropdown-articles"
                data-bs-toggle="dropdown"
                aria-expanded="false"
                aria-haspopup="true"
              >
                Articles
              </button>
              <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_clustering.html"
                    >Clustering</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_data.html"
                    >Data Input</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_decoupleR.html"
                    >DecoupleR integration</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_example.html"
                    >Example</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_faq.html"
                    >FAQ</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_LR.html"
                    >Ligand-Receptor Analysis</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_utils.html"
                    >Workflow Utilities</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_visualization.html"
                    >Visualization</a
                  >
                </li>
              </ul>
            </li>
          </ul>
          <ul class="navbar-nav">
            <li class="nav-item">
              <form class="form-inline" role="search">
                <input
                  class="form-control"
                  type="search"
                  name="search-input"
                  id="search-input"
                  autocomplete="off"
                  aria-label="Search site"
                  placeholder="Search for"
                  data-search-index="../search.json"
                />
              </form>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div class="container template-article">
      <div class="row">
        <main id="main" class="col-md-9">
          <div class="page-header">
            <img src="../logo.png" class="logo" alt="" />
            <h1>Workflow Utilities</h1>

            <div class="d-none name"><code>spacedeconv_utils.Rmd</code></div>
          </div>

          <p>
            <code>spacedeconv</code> offers a variety of workflow helper
            functions that streamline the overall analysis process. In the
            following we will give an overview over the available functions.
          </p>
          <ol style="list-style-type: decimal">
            <li><code>preprocess</code></li>
            <li><code>normalize</code></li>
            <li><code>print_info</code></li>
            <li><code>available_results</code></li>
            <li><code>aggregate_results</code></li>
            <li><code>addCustomAnnotation</code></li>
            <li><code>annotate_spots</code></li>
            <li><code>scale_cell_counts</code></li>
            <li><code>subsetSCE</code></li>
            <li><code>subsetSPE</code></li>
          </ol>
          <div class="sourceCode" id="cb1">
            <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://omnideconv.github.io/spacedeconv">spacedeconv</a></span><span class="op">)</span></span></code></pre>
          </div>
          <pre><code><span><span class="co">## → Setting up spacedeconv environment..</span></span></code></pre>
          <pre><code><span><span class="co">## → Using conda environment 'r-omnideconv'</span></span></code></pre>
          <div class="section level2">
            <h2 id="preprocess">
              1. <code>preprocess</code
              ><a class="anchor" aria-label="anchor" href="#preprocess"></a>
            </h2>
            <p>
              The function can be used to preprocess single-cell or spatial
              datasets. The cuts of low and high UMI observations, removes noisy
              expression and performs additional checks to streamline the
              deconvolution analysis. The functions takes a
              SingleCellExperiment, AnnData or Seurat and returns a processed
              SingleCellExperiment. <code>min_umi</code> or
              <code>max_umi</code> parameters can be set to improve data
              quality. The assay can be selected with the
              <code>assay</code> parameter. Additionally Mitochondria Genes can
              be removed by setting <code>remove_mito=TRUE</code>.
            </p>
            <div class="sourceCode" id="cb4">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_3"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">single_cell_data_3</span>, min_umi <span class="op">=</span> <span class="fl">500</span>, assay <span class="op">=</span> <span class="st">"counts"</span>, remove_mito <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[84ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Removing 8 observations with umi count below threshold</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Removed 8 observations with umi count below threshold <span style="color: #B2B2B2;">[1.6s]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Removing 5862 variables with all zero expression</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Removed 5862 variables with all zero expression <span style="color: #B2B2B2;">[921ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Removing 13 mitochondria genes</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Removed 13 mitochondria genes <span style="color: #B2B2B2;">[843ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers</span></span>
<span><span class="co">## <span style="color: #BBBB00;">!</span> Warning: ENSEMBL identifiers detected in gene names</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BBBB;">ℹ</span> Consider using Gene Names for first-generation deconvolution tools</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BB00;">✔</span> Finished Preprocessing <span style="color: #B2B2B2;">[10ms]</span></span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">spatial_data_3</span>, min_umi <span class="op">=</span> <span class="fl">500</span>, assay <span class="op">=</span> <span class="st">"counts"</span>, remove_mito <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[35ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Removing 137 observations with umi count below threshold</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Removed 137 observations with umi count below threshold <span style="color: #B2B2B2;">[128ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Removing 13049 variables with all zero expression</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Removed 13049 variables with all zero expression <span style="color: #B2B2B2;">[147ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Removing 13 mitochondria genes</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Removed 13 mitochondria genes <span style="color: #B2B2B2;">[120ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers</span></span>
<span><span class="co">## <span style="color: #BBBB00;">!</span> Warning: ENSEMBL identifiers detected in gene names</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BBBB;">ℹ</span> Consider using Gene Names for first-generation deconvolution tools</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BB00;">✔</span> Finished Preprocessing <span style="color: #B2B2B2;">[11ms]</span></span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="normalize">
              2. <code>normalize</code
              ><a class="anchor" aria-label="anchor" href="#normalize"></a>
            </h2>
            <p>
              You can scale and normalize your single-cell or spatial data by
              calling the <code>normalize function</code>. The function takes a
              <code>method</code> parameter where <code>cpm</code>or
              <code>logcpm</code>can be selected. The normalized data is stored
              as an additional assay in the object.
            </p>
            <div class="sourceCode" id="cb5">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">sce</span>, method <span class="op">=</span> <span class="st">"cpm"</span>, assay <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="co">## ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[20ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Normalizing using cpm</span></span>
<span><span class="co">## Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size</span></span>
<span><span class="co">## 1.4 GiB</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Finished normalization using cpm <span style="color: #B2B2B2;">[3.9s]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please note the normalization is stored in an additional assay</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spe</span>, method <span class="op">=</span> <span class="st">"cpm"</span>, assay <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="co">## ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[17ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Normalizing using cpm</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Finished normalization using cpm <span style="color: #B2B2B2;">[395ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please note the normalization is stored in an additional assay</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="print_info">
              3. <code>print_info</code
              ><a class="anchor" aria-label="anchor" href="#print_info"></a>
            </h2>
            <p>
              You can obtain additional info about your dataset by calling
              <code>print_info</code>.
            </p>
            <div class="sourceCode" id="cb6">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/print_info.html">print_info</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Single Cell</span></span>
<span><span class="co">## Assays: <span style="color: #0000BB;">"counts"</span> and <span style="color: #0000BB;">"cpm"</span></span></span>
<span><span class="co">## Genes: <span style="color: #0000BB;">23858</span></span></span>
<span><span class="co">## → without expression: <span style="color: #0000BB;">0</span> (0%)</span></span>
<span><span class="co">## Cells: <span style="color: #0000BB;">7978</span></span></span>
<span><span class="co">## → without expression: <span style="color: #0000BB;">0</span> (0%)</span></span>
<span><span class="co">## Umi count range: <span style="color: #0000BB;">447</span> - <span style="color: #0000BB;">74244</span></span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Rownames set</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Colnames set</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="available_results">
              4. <code>available_results</code
              ><a
                class="anchor"
                aria-label="anchor"
                href="#available_results"
              ></a>
            </h2>
            <p>
              You can check what deconvolution results and additional annotation
              is available in your data by calling
              <code>available_resutls</code>. You can set the
              <code>method</code> parameter to the name of a deconvolution tool
              to further filter the results if many quantifications where
              performed.
            </p>
            <div class="sourceCode" id="cb7">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># "deconv" contains DWLS results</span></span>
<span><span class="fu"><a href="../reference/available_results.html">available_results</a></span><span class="op">(</span><span class="va">deconv</span><span class="op">)</span></span></code></pre>
            </div>
            <pre><code><span><span class="co">## [1] "spatialdwls_B.cells"           "spatialdwls_CAFs"             </span></span>
<span><span class="co">## [3] "spatialdwls_Cancer.Epithelial" "spatialdwls_Endothelial"      </span></span>
<span><span class="co">## [5] "spatialdwls_Myeloid"           "spatialdwls_Normal.Epithelial"</span></span>
<span><span class="co">## [7] "spatialdwls_Plasmablasts"      "spatialdwls_PVL"              </span></span>
<span><span class="co">## [9] "spatialdwls_T.cells"</span></span></code></pre>
          </div>
          <div class="section level2">
            <h2 id="aggregate_results">
              5. <code>aggregate_results</code
              ><a
                class="anchor"
                aria-label="anchor"
                href="#aggregate_results"
              ></a>
            </h2>
            <p>
              You can aggregate fine-grained deconvolution results to a single
              value by providing a list of deconvolution result names to the
              <code>cell_types</code> parameter. You can additionally set a new
              <code>name</code> and you have the option to
              <code>remove</code> the original fine-grained columns and just
              keep the aggregation.
            </p>
            <div class="sourceCode" id="cb9">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/aggregate_results.html">aggregate_results</a></span><span class="op">(</span><span class="va">deconv</span>, cell_types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"spatialdwls_Cancer.Epithelial"</span>, <span class="st">"spatialdwls_Normal.Epithelial"</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"spatialdwls_Epithelial"</span>, remove <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[9ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Aggregating cell types</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> Aggregated cell types <span style="color: #B2B2B2;">[12ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## class: SpatialExperiment </span></span>
<span><span class="co">## dim: 23529 1185 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(2): counts cpm</span></span>
<span><span class="co">## rownames(23529): AL627309.1 AL627309.5 ... AC007325.4 AC007325.2</span></span>
<span><span class="co">## rowData names(2): symbol ensembl</span></span>
<span><span class="co">## colnames(1185): AAACAATCTACTAGCA-1 AAACACCAATAACTGC-1 ...</span></span>
<span><span class="co">##   TTGTTTCATTAGTCTA-1 TTGTTTGTGTAAATTC-1</span></span>
<span><span class="co">## colData names(12): in_tissue array_row ... spatialdwls_Myeloid</span></span>
<span><span class="co">##   spatialdwls_Epithelial</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres</span></span>
<span><span class="co">## imgData names(4): sample_id image_id data scaleFactor</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="addcustomannotation">
              6. <code>addCustomAnnotation</code
              ><a
                class="anchor"
                aria-label="anchor"
                href="#addcustomannotation"
              ></a>
            </h2>
            <p>
              This function helps adding a custom annotation vector to a
              SpatialExperiment object.
            </p>
            <div class="sourceCode" id="cb10">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># newAnnotation is a vector containing custom annotation for each spot</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCustomAnnotation.html">addCustomAnnotation</a></span><span class="op">(</span><span class="va">spe</span>, columnName <span class="op">=</span> <span class="st">"ManualAnnotation"</span>, values <span class="op">=</span> <span class="va">new_annotation</span><span class="op">)</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="annotate_spots">
              7. <code>annotate_spots</code
              ><a class="anchor" aria-label="anchor" href="#annotate_spots"></a>
            </h2>
            <p>
              This function is able to annotate spots with TRUE / FALSE if you
              want to classify a specific subgroup of spots. It takes a list of
              spots that should be classified as TRUE, setting all other spots
              to FALSE.
            </p>
            <div class="sourceCode" id="cb11">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># spots is a list of spot names.</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_spots.html">annotate_spots</a></span><span class="op">(</span><span class="va">spe</span>, <span class="va">spots</span>, value_pos <span class="op">=</span> <span class="cn">TRUE</span>, value_neg <span class="op">=</span> <span class="cn">FALSE</span>, name <span class="op">=</span> <span class="st">"customAnnotation"</span><span class="op">)</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="scale_cell_counts">
              8. <code>scale_cell_counts</code
              ><a
                class="anchor"
                aria-label="anchor"
                href="#scale_cell_counts"
              ></a>
            </h2>
            <p>
              Most deconvolution tools compute relative cell fractions for
              spots. If you have cell counts for each spot you can scale the
              relative values to absolute cell counts using this function. The
              input parameters are the column name that should be scaled
              <code>value</code> and a vector of absolute cell counts for each
              spot <code>cell_counts</code>. You can also set a new
              <code>resName</code>.
            </p>
            <div class="sourceCode" id="cb12">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cell_counts_per_spot contains spot level absolute cell counts</span></span>
<span><span class="va">spe_absolute</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scale_cell_counts.html">scale_cell_counts</a></span><span class="op">(</span><span class="va">spe</span>, value <span class="op">=</span> <span class="st">"spatialdwls_B.cells"</span>, cell_counts <span class="op">=</span> <span class="va">cell_counts_per_spot</span>, resName <span class="op">=</span> <span class="st">"BCellsAbsolute"</span><span class="op">)</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="subsetsce">
              9. <code>subsetSCE</code
              ><a class="anchor" aria-label="anchor" href="#subsetsce"></a>
            </h2>
            <p>
              To improve resource requirements for deconvolution computation you
              can reduce your input scRNA-seq reference size by subsetting. The
              functions requires your input <code>sce</code> object, the column
              name containing the cell-type annotation
              <code>cell_type_col</code>. You can specify the subsetting
              scenario <code>scenario</code> as one of “mirror” or “even”. The
              mirror scenario keeps the same cell-type proportions as in the
              input data but reduces the overall cell number. The even scenario
              selects the same number of cells for each cell-type. Specify the
              number of cells you want after subsetting using the
              <code>ncells</code> parameter. In case <code>notEnough</code>cells
              are available for a cell-type to match the required number
              according to the scenario you can set this parameter to “asis” to
              keep all remaining cells or “remove” the cell-type completely.
            </p>
            <div class="sourceCode" id="cb13">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetSCE.html">subsetSCE</a></span><span class="op">(</span><span class="va">sce</span>, cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>, scenario <span class="op">=</span> <span class="st">"mirror"</span>, ncells <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span>
<span><span class="co">## ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Set seed to 12345</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> testing parameter<span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[16ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> extracting up to 500 cells</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> extracting up to 500 cells <span style="color: #B2B2B2;">[183ms]</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> extracted 501 cells</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> extracted 501 cells <span style="color: #B2B2B2;">[20ms]</span></span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="subsetspe">
              10. <code>subsetSPE</code
              ><a class="anchor" aria-label="anchor" href="#subsetspe"></a>
            </h2>
            <p>
              When not working with the full Spatial data you can opt for using
              a subset of the spatial slide. This function take a
              <code>colRange</code> and <code>rowRange</code> vector containing
              the pixel coordinates of the fullres image to cut subsets of the
              provided spatial data.
            </p>
            <div class="sourceCode" id="cb14">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetSPE.html">subsetSPE</a></span><span class="op">(</span><span class="va">spe</span>, colRange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1000</span><span class="op">)</span>, rowRange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1000</span><span class="op">)</span><span class="op">)</span></span></code></pre>
            </div>
          </div>
        </main>
        <aside class="col-md-3">
          <nav id="toc" aria-label="Table of contents">
            <h2>On this page</h2>
          </nav>
        </aside>
      </div>

      <footer>
        <div class="pkgdown-footer-left">
          <p>Developed by Constantin Zackl.</p>
        </div>

        <div class="pkgdown-footer-right">
          <p>
            Site built with
            <a href="https://pkgdown.r-lib.org/" class="external-link"
              >pkgdown</a
            >
            2.1.3.
          </p>
        </div>
      </footer>
    </div>
  </body>
</html>
