<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started • spacedeconv</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/spacedeconv.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/spacedeconv_clustering.html">Clustering</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_data.html">Data Input</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_decoupleR.html">DecoupleR integration</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_example.html">Example</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_faq.html">FAQ</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_LR.html">Ligand-Receptor Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_utils.html">Workflow Utilities</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_visualization.html">Visualization</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Getting started</h1>
            
      

      <div class="d-none name"><code>spacedeconv.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="spacedeconv">spacedeconv<a class="anchor" aria-label="anchor" href="#spacedeconv"></a>
</h2>
<p>spacedeconv is a unified interface to 20 deconvolution tools with
focus on spatial transcriptomics datasets. In total 17 second-generation
deconvolution tools are included, enabling deconvolution of any cell
types when single-cell reference data is available. Additionally 10
first-generation tools, which are focussing on deconvolution of immune
cells, are available as well as 4 first-generation methods optimised for
mouse data. These methods don’t require scRNA-seq data to perform
deconvolution as they utilize precomputed and verified signatures.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>There are two ways to install <code>spacedeconv</code>:</p>
<ul>
<li>The <em>minimal</em> installation installs only the dependencies
required for the basic functionalities. All deconvolution methods need
to be installed on-demand.</li>
<li>The <em>complete</em> installtation installs all dependencies
including all deconvolution methods. This may take a considerable
time.</li>
</ul>
<p>Since not all dependencies are on CRAN or Bioconductor,
<code>spacedeconv</code> is available from GitHub only. We recommend
installing trough the pak package manager:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install the pak package manager</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"pak"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># recommended installation, deconvolution tools are installed on-demand</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/spacedeconv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># full installation including all deconvolution tools</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"omnideconv/spacedeconv"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-requirements">Data requirements<a class="anchor" aria-label="anchor" href="#data-requirements"></a>
</h2>
<p>spacedeconv offers convenient access to perform first- and
second-generation deconvolution on spatial transcriptomics datasets.
While deconvolution can be performed directly with first-generation
methods, second-generation algorithms require an additional annotated
single-cell reference.</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html" class="external-link">SpatialExperiment</a></em>,
will be deconvoluted</li>
<li>
<em><a href="https://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html" class="external-link">SingleCellExperiment</a></em>
(recommended), <em><a href="https://anndata.dynverse.org/" class="external-link">anndata</a></em> or <em><a href="https://satijalab.org/seurat/" class="external-link">Seurat</a></em> containing cell
type information</li>
</ul>
<p>You can load 10X Visium Data by providing the <a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger" class="external-link">spaceranger</a>
output folder. It is further possible to run spacedeconv with manually
created SpatialExperiments. See the SpatialExperiment <a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html" class="external-link">Documentation</a>
for further details.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">SpatialExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">read10xVisium</a></span><span class="op">(</span><span class="st">"path_to_directory"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>The core modules of spacedeconv are used to build a reference
signature and deconvolute spatial transcriptomics data. We further
provide convenient functions for normalization and visualization. A full
list of deconvolution algorithms can be accessed by
<code><a href="../reference/deconvolution_methods.html">spacedeconv::deconvolution_methods</a></code> or in the <a href="spacedeconv_faq.html">FAQ</a>.</p>
<div class="section level4">
<h4 id="normalization">1. Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h4>
<p>spacedeconv offers a function to normalize spatial expression data.
The normalization is saved in an additional assay, so make sure to use
the correct data during deconvolution by providing the desired assay
with the parameters <code>assay_sc</code> and <code>assay_sp</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spe</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># specify expression assay to use</span></span>
<span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span><span class="va">spe</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"quantiseq"</span>,</span>
<span>  assay_sp <span class="op">=</span> <span class="st">"cpm"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="building-a-reference-signature">2. Building a reference Signature<a class="anchor" aria-label="anchor" href="#building-a-reference-signature"></a>
</h4>
<p>All second-generation deconvolution tools require a cell-type
specific reference signature which can be calculated with
<code><a href="../reference/build_model.html">build_model()</a></code> function. To build a signature annotated
single-cell sequencing data is required.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  <span class="va">single_cell_object</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spotlight"</span>,</span>
<span>  assay_sc <span class="op">=</span> <span class="st">"cpm"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="deconvolution">3. Deconvolution<a class="anchor" aria-label="anchor" href="#deconvolution"></a>
</h4>
<p>This function deconvolutes spatial transcriptomics dataset using one
of spacedeconvs 20 deconvolution tools. Since first-generation
deconvolution tools contain predefined expression signatures these tools
can be used directly on the spatial data. You do not need to provide a
signature for this case. spacedeconv returns a SpatialExperiment object
annotated with deconvolution results.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># save the results to an annotated SpatialExperiment</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  <span class="va">spatial_object</span>,</span>
<span>  <span class="va">signature</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spotlight"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># return deconvolution results in table form</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  <span class="va">spatial_object</span>,</span>
<span>  <span class="va">signature</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spotlight"</span>,</span>
<span>  return_object <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualization">4. Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h4>
<p>We offer 6 visualization function aiming different insight of the
data. A full example of each visualization option is available in the
visualization <a href="spacedeconv_visualization.html">Vignette</a>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># sample does refer to the first column of ColData(spe)</span></span>
<span><span class="co"># for cell_type input a celltype present in the deconvolution result</span></span>
<span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">spe</span>, cell_type <span class="op">=</span> <span class="st">"spotlight_B.cells"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># umi count</span></span>
<span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_umi_count.html">plot_umi_count</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
