<!doctype html>
<!-- Generated by pkgdown: do not edit by hand -->
<html lang="en">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
    />
    <title>Example • spacedeconv</title>
    <!-- favicons -->
    <link
      rel="icon"
      type="image/png"
      sizes="96x96"
      href="../favicon-96x96.png"
    />
    <link rel="icon" type="”image/svg+xml”" href="../favicon.svg" />
    <link
      rel="apple-touch-icon"
      sizes="180x180"
      href="../apple-touch-icon.png"
    />
    <link rel="icon" sizes="any" href="../favicon.ico" />
    <link rel="manifest" href="../site.webmanifest" />
    <script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
    />
    <link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet" />
    <script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script>
    <link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
    <link
      href="../deps/font-awesome-6.5.2/css/v4-shims.min.css"
      rel="stylesheet"
    />
    <script src="../deps/headroom-0.11.0/headroom.min.js"></script>
    <script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script>
    <script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script>
    <script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script>
    <script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script>
    <script src="../deps/search-1.0.0/fuse.min.js"></script>
    <script src="../deps/search-1.0.0/mark.min.js"></script>
    <!-- pkgdown -->
    <script src="../pkgdown.js"></script>
    <meta property="og:title" content="Example" />
  </head>
  <body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>

    <nav
      class="navbar navbar-expand-lg fixed-top bg-light"
      data-bs-theme="light"
      aria-label="Site navigation"
    >
      <div class="container">
        <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

        <small
          class="nav-text text-muted me-auto"
          data-bs-toggle="tooltip"
          data-bs-placement="bottom"
          title=""
          >0.1.1</small
        >

        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbar"
          aria-controls="navbar"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>

        <div id="navbar" class="collapse navbar-collapse ms-3">
          <ul class="navbar-nav me-auto">
            <li class="nav-item">
              <a class="nav-link" href="../articles/spacedeconv.html"
                >Get started</a
              >
            </li>
            <li class="nav-item">
              <a class="nav-link" href="../reference/index.html">Reference</a>
            </li>
            <li class="active nav-item dropdown">
              <button
                class="nav-link dropdown-toggle"
                type="button"
                id="dropdown-articles"
                data-bs-toggle="dropdown"
                aria-expanded="false"
                aria-haspopup="true"
              >
                Articles
              </button>
              <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_clustering.html"
                    >Clustering</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_data.html"
                    >Data Input</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_decoupleR.html"
                    >DecoupleR integration</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_example.html"
                    >Example</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_faq.html"
                    >FAQ</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_LR.html"
                    >Ligand-Receptor Analysis</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_utils.html"
                    >Workflow Utilities</a
                  >
                </li>
                <li>
                  <a
                    class="dropdown-item"
                    href="../articles/spacedeconv_visualization.html"
                    >Visualization</a
                  >
                </li>
              </ul>
            </li>
          </ul>
          <ul class="navbar-nav">
            <li class="nav-item">
              <form class="form-inline" role="search">
                <input
                  class="form-control"
                  type="search"
                  name="search-input"
                  id="search-input"
                  autocomplete="off"
                  aria-label="Search site"
                  placeholder="Search for"
                  data-search-index="../search.json"
                />
              </form>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div class="container template-article">
      <div class="row">
        <main id="main" class="col-md-9">
          <div class="page-header">
            <img src="../logo.png" class="logo" alt="" />
            <h1>Example</h1>

            <div class="d-none name"><code>spacedeconv_example.Rmd</code></div>
          </div>

          <p>
            spacedeconv is a unified interface for the deconvolution of spatial
            transcriptomics data. In total spacedeconv gives access to 20
            deconvolution methods:
          </p>
          <pre><code><span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:spacedeconv':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     normalize</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="co">#&gt;             RCTD        SPOTlight             CARD      spatialDWLS </span></span>
<span><span class="co">#&gt;           "rctd"      "spotlight"           "card"    "spatialdwls" </span></span>
<span><span class="co">#&gt;    cell2location              DOT       MCPcounter             EPIC </span></span>
<span><span class="co">#&gt;  "cell2location"            "dot"    "mcp_counter"           "epic" </span></span>
<span><span class="co">#&gt;        quanTIseq            xCell        CIBERSORT CIBERSORT (abs.) </span></span>
<span><span class="co">#&gt;      "quantiseq"          "xcell"      "cibersort"  "cibersort_abs" </span></span>
<span><span class="co">#&gt;            TIMER     ConsensusTME             ABIS         ESTIMATE </span></span>
<span><span class="co">#&gt;          "timer"  "consensus_tme"           "abis"       "estimate" </span></span>
<span><span class="co">#&gt;      mMCPcounter        seqImmuCC              DCQ             BASE </span></span>
<span><span class="co">#&gt;   "mmcp_counter"      "seqimmucc"            "dcq"           "base"</span></span></code></pre>
          <div class="section level2">
            <h2 id="sample-data">
              Sample Data<a
                class="anchor"
                aria-label="anchor"
                href="#sample-data"
              ></a>
            </h2>
            <p>
              For this tutorial we will use one of spacedeconvs sample dataset
              and the deconvolution algorithm “SCDC”.
            </p>
            <div class="sourceCode" id="cb2">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_3"</span><span class="op">)</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="preprocessing-and-normalization">
              Preprocessing and Normalization<a
                class="anchor"
                aria-label="anchor"
                href="#preprocessing-and-normalization"
              ></a>
            </h2>
            <p>
              To get a first insight into the data we type the dataset name into
              the terminal:
            </p>
            <div class="sourceCode" id="cb3">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 29733 7986 </span></span>
<span><span class="co">#&gt; metadata(1): Samples</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(29733): RP11-34P13.7 FO538757.3 ... KRTAP9-2 IGLVIV-66-1</span></span>
<span><span class="co">#&gt; rowData names(1): ID</span></span>
<span><span class="co">#&gt; colnames(7986): CID44971_AAGCCGCCACGCATCG CID44971_AAGGAGCTCAACACAC ...</span></span>
<span><span class="co">#&gt;   CID44971_TTTCCTCCAAGCCATT CID44971_TTTGGTTGTATCACCA</span></span>
<span><span class="co">#&gt; colData names(10): Sample Barcode ... celltype_minor celltype_major</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre>
            </div>
            <p>
              We can see that the single-cell data is available as a
              <a
                href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html"
                class="external-link"
                >SingleCellExperiment</a
              >. In total this dataset includes expression values from 29733
              genes and 7986 cells. It can further be seen that cell type
              information is available in <code>colData</code> where cell
              related annotation is stored.
            </p>
            <p>
              As the next step we normalize the data as
              <code>cpm</code> (“Counts Per Million”). The normalization is
              stored as an additional assay in the single-cell object.
            </p>
            <div class="sourceCode" id="cb4">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[65ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 8 observations with umi count below threshold</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 8 observations with umi count below threshold <span style="color: #B2B2B2;">[1.6s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 5862 variables with all zero expression</span></span>
<span><span class="co">#&gt; Warning in spacedeconv::preprocess(single_cell_data_3): There are 13</span></span>
<span><span class="co">#&gt; mitochondrial genes present. Consider removing them.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 5862 variables with all zero expression <span style="color: #B2B2B2;">[908ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Warning: ENSEMBL identifiers detected in gene names</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BBBB;">ℹ</span> Consider using Gene Names for first-generation deconvolution tools</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BB00;">✔</span> Finished Preprocessing <span style="color: #B2B2B2;">[9ms]</span></span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">spatial_data_3</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[34ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 137 observations with umi count below threshold</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 137 observations with umi count below threshold <span style="color: #B2B2B2;">[128ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 13049 variables with all zero expression</span></span>
<span><span class="co">#&gt; Warning in spacedeconv::preprocess(spatial_data_3): There are 13 mitochondrial</span></span>
<span><span class="co">#&gt; genes present. Consider removing them.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 13049 variables with all zero expression <span style="color: #B2B2B2;">[162ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Warning: ENSEMBL identifiers detected in gene names</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BBBB;">ℹ</span> Consider using Gene Names for first-generation deconvolution tools</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BB00;">✔</span> Finished Preprocessing <span style="color: #B2B2B2;">[11ms]</span></span></span>
<span></span>
<span><span class="va">single_cell_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">single_cell_data_3</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[20ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Normalizing using cpm</span></span>
<span><span class="co">#&gt; Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size</span></span>
<span><span class="co">#&gt; 1.4 GiB</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Finished normalization using cpm <span style="color: #B2B2B2;">[3.8s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please note the normalization is stored in an additional assay</span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spatial_data_3</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[17ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Normalizing using cpm</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Finished normalization using cpm <span style="color: #B2B2B2;">[376ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please note the normalization is stored in an additional assay</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="build-model">
              Build Model<a
                class="anchor"
                aria-label="anchor"
                href="#build-model"
              ></a>
            </h2>
            <p>
              To build a model we need to provide a single cell expression
              object annotated with cell type information. Let’s have a look
              into the object to see which annotation is available.
            </p>
            <div class="sourceCode" id="cb5">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Sample"          "Barcode"         "orig.ident"      "nCount_RNA"     </span></span>
<span><span class="co">#&gt;  [5] "nFeature_RNA"    "percent.mito"    "subtype"         "celltype_subset"</span></span>
<span><span class="co">#&gt;  [9] "celltype_minor"  "celltype_major"</span></span></code></pre>
            </div>
            <p>
              There are three different resolutions of cell type annotation
              available. For this tutorial we will choose
              <code>"celltype_major"</code> as cell type annotation which
              includes 9 different cell-types.
            </p>
            <p>
              Next we can calculate a signature using the annotated single-data
              and the deconvolution tool “spatialDWLS”.
            </p>
            <div class="sourceCode" id="cb6">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spatialdwls"</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; class of selected matrix:  dgCMatrix </span></span>
<span><span class="co">#&gt; [1] "finished runPCA_factominer, method == factominer"</span></span></code></pre>
            </div>
            <p>
              The signature contains expression values for all celltypes and a
              subset of distinctly expressed genes. The following example shows
              the first 10 rows of the signature calculated in the previous
              step.
            </p>
            <div class="sourceCode" id="cb7">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">signature</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></code></pre>
            </div>
            <table class="table">
              <colgroup>
                <col width="8%" />
                <col width="10%" />
                <col width="7%" />
                <col width="7%" />
                <col width="7%" />
                <col width="11%" />
                <col width="7%" />
                <col width="7%" />
                <col width="16%" />
                <col width="16%" />
              </colgroup>
              <thead>
                <tr class="header">
                  <th align="left"></th>
                  <th align="right">Endothelial</th>
                  <th align="right">CAFs</th>
                  <th align="right">PVL</th>
                  <th align="right">B-cells</th>
                  <th align="right">Plasmablasts</th>
                  <th align="right">T-cells</th>
                  <th align="right">Myeloid</th>
                  <th align="right">Cancer Epithelial</th>
                  <th align="right">Normal Epithelial</th>
                </tr>
              </thead>
              <tbody>
                <tr class="odd">
                  <td align="left">TNFRSF18</td>
                  <td align="right">0.0509</td>
                  <td align="right">0.0464</td>
                  <td align="right">0.0110</td>
                  <td align="right">0.0244</td>
                  <td align="right">0.8333</td>
                  <td align="right">0.8543</td>
                  <td align="right">0.0441</td>
                  <td align="right">0.0638</td>
                  <td align="right">0.0490</td>
                </tr>
                <tr class="even">
                  <td align="left">TNFRSF4</td>
                  <td align="right">0.5926</td>
                  <td align="right">0.1443</td>
                  <td align="right">0.0220</td>
                  <td align="right">0.0379</td>
                  <td align="right">0.3958</td>
                  <td align="right">0.8735</td>
                  <td align="right">0.1000</td>
                  <td align="right">0.0962</td>
                  <td align="right">0.0123</td>
                </tr>
                <tr class="odd">
                  <td align="left">MXRA8</td>
                  <td align="right">0.0741</td>
                  <td align="right">4.7835</td>
                  <td align="right">0.6813</td>
                  <td align="right">0.0081</td>
                  <td align="right">0.0208</td>
                  <td align="right">0.0078</td>
                  <td align="right">0.0103</td>
                  <td align="right">0.0268</td>
                  <td align="right">0.3556</td>
                </tr>
                <tr class="even">
                  <td align="left">TNFRSF9</td>
                  <td align="right">0.0324</td>
                  <td align="right">0.0241</td>
                  <td align="right">0.0000</td>
                  <td align="right">0.0027</td>
                  <td align="right">0.0208</td>
                  <td align="right">0.4319</td>
                  <td align="right">0.0735</td>
                  <td align="right">0.0157</td>
                  <td align="right">0.0095</td>
                </tr>
                <tr class="odd">
                  <td align="left">C1QA</td>
                  <td align="right">0.0185</td>
                  <td align="right">0.0326</td>
                  <td align="right">0.0110</td>
                  <td align="right">0.0190</td>
                  <td align="right">0.0000</td>
                  <td align="right">0.0128</td>
                  <td align="right">12.6912</td>
                  <td align="right">0.0649</td>
                  <td align="right">0.0259</td>
                </tr>
                <tr class="even">
                  <td align="left">C1QB</td>
                  <td align="right">0.0278</td>
                  <td align="right">0.0258</td>
                  <td align="right">0.0659</td>
                  <td align="right">0.0379</td>
                  <td align="right">0.0208</td>
                  <td align="right">0.0179</td>
                  <td align="right">17.7029</td>
                  <td align="right">0.0738</td>
                  <td align="right">0.0341</td>
                </tr>
                <tr class="odd">
                  <td align="left">RPL11</td>
                  <td align="right">24.0417</td>
                  <td align="right">20.5979</td>
                  <td align="right">16.4066</td>
                  <td align="right">20.2818</td>
                  <td align="right">15.8125</td>
                  <td align="right">13.7076</td>
                  <td align="right">24.2588</td>
                  <td align="right">54.4430</td>
                  <td align="right">69.8583</td>
                </tr>
                <tr class="even">
                  <td align="left">STMN1</td>
                  <td align="right">2.1528</td>
                  <td align="right">0.6271</td>
                  <td align="right">0.3077</td>
                  <td align="right">0.2629</td>
                  <td align="right">0.6667</td>
                  <td align="right">1.3483</td>
                  <td align="right">2.5529</td>
                  <td align="right">36.6074</td>
                  <td align="right">1.3215</td>
                </tr>
                <tr class="odd">
                  <td align="left">SH3BGRL3</td>
                  <td align="right">4.3796</td>
                  <td align="right">2.3385</td>
                  <td align="right">1.7802</td>
                  <td align="right">3.5339</td>
                  <td align="right">3.5417</td>
                  <td align="right">7.4338</td>
                  <td align="right">18.4397</td>
                  <td align="right">9.7427</td>
                  <td align="right">2.6621</td>
                </tr>
                <tr class="even">
                  <td align="left">CD52</td>
                  <td align="right">0.2130</td>
                  <td align="right">0.1237</td>
                  <td align="right">0.1319</td>
                  <td align="right">5.6775</td>
                  <td align="right">0.9792</td>
                  <td align="right">3.3556</td>
                  <td align="right">3.6662</td>
                  <td align="right">0.2125</td>
                  <td align="right">0.0899</td>
                </tr>
              </tbody>
            </table>
          </div>
          <div class="section level2">
            <h2 id="deconvolution">
              Deconvolution<a
                class="anchor"
                aria-label="anchor"
                href="#deconvolution"
              ></a>
            </h2>
            <p>
              To deconvolute the Visium slide we use the signature we just
              calculated. Depending on the selected deconvolution tool further
              parameters need to be passed, in this case
              <code>cell_type_col</code> and <code>batch_id_col</code> as well
              as the scRNA-seq reference dataset.
            </p>
            <div class="sourceCode" id="cb8">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deconv</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  spatial_obj <span class="op">=</span> <span class="va">spatial_data_3</span>,</span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spatialdwls"</span>,</span>
<span>  signature <span class="op">=</span> <span class="va">signature</span>,</span>
<span>  assay_sp <span class="op">=</span> <span class="st">"cpm"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; class of selected matrix:  dgCMatrix </span></span>
<span><span class="co">#&gt; [1] "finished runPCA_factominer, method == factominer"</span></span></code></pre>
            </div>
          </div>
          <div class="section level2">
            <h2 id="visualization">
              Visualization<a
                class="anchor"
                aria-label="anchor"
                href="#visualization"
              ></a>
            </h2>
            <p>
              The deconvolution results are stored in the spatial object to
              simplify visualizations. spacedeconv offers 6 visualization
              functions with different scopes, further explained in the
              visualization
              <a href="spacedeconv_visualization.html">vignette</a>.
            </p>
            <p>
              We can see the available deconvolution results by using
              <code>available_results(deconv)</code> or by accessing the objects
              <code>colData</code>:
            </p>
            <div class="sourceCode" id="cb9">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/available_results.html">available_results</a></span><span class="op">(</span><span class="va">deconv</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "spatialdwls_B.cells"           "spatialdwls_CAFs"             </span></span>
<span><span class="co">#&gt; [3] "spatialdwls_Cancer.Epithelial" "spatialdwls_Endothelial"      </span></span>
<span><span class="co">#&gt; [5] "spatialdwls_Myeloid"           "spatialdwls_Normal.Epithelial"</span></span>
<span><span class="co">#&gt; [7] "spatialdwls_Plasmablasts"      "spatialdwls_PVL"              </span></span>
<span><span class="co">#&gt; [9] "spatialdwls_T.cells"</span></span></code></pre>
            </div>
            <p>
              For each Spot the estimated cell fraction is available as object
              annotation and can be used for visualizations.
            </p>
            <div class="sourceCode" id="cb10">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot all available results: provide the tool as parameter</span></span>
<span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_spatial.html">plot_spatial</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  result <span class="op">=</span> <span class="st">"spatialdwls_B.cells"</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  smooth <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  title_size <span class="op">=</span> <span class="fl">12</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">spacedeconv</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: `label` cannot be a <span style="color: #0000BB;">&lt;ggplot2::element_blank&gt;</span> object.</span></span></code></pre>
            </div>
            <p>
              <img
                src="spacedeconv_example_files/figure-html/visualization-1.png"
                width="1152"
              />
            </p>
            <div class="sourceCode" id="cb11">
              <pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">#&gt; Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Rocky Linux 8.10 (Green Obsidian)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS/LAPACK: /gpfs/gpfs1/scratch/c9881013/.conda_envs/r-omnideconv/lib/libmkl_rt.so;  LAPACK version 3.8.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Vienna</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] SpatialExperiment_1.12.0    SingleCellExperiment_1.24.0</span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             </span></span>
<span><span class="co">#&gt;  [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        </span></span>
<span><span class="co">#&gt;  [7] IRanges_2.36.0              S4Vectors_0.40.2           </span></span>
<span><span class="co">#&gt;  [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      </span></span>
<span><span class="co">#&gt; [11] matrixStats_1.5.0           spacedeconv_0.1.1          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] later_1.4.4               bitops_1.0-9             </span></span>
<span><span class="co">#&gt;   [3] tibble_3.3.0              cellranger_1.1.0         </span></span>
<span><span class="co">#&gt;   [5] lifecycle_1.0.4           sf_1.0-20                </span></span>
<span><span class="co">#&gt;   [7] rstatix_0.7.2             edgeR_4.0.16             </span></span>
<span><span class="co">#&gt;   [9] lattice_0.22-7            backports_1.5.0          </span></span>
<span><span class="co">#&gt;  [11] magrittr_2.0.4            limma_3.58.1             </span></span>
<span><span class="co">#&gt;  [13] sass_0.4.10               rmarkdown_2.30           </span></span>
<span><span class="co">#&gt;  [15] jquerylib_0.1.4           yaml_2.3.10              </span></span>
<span><span class="co">#&gt;  [17] remotes_2.5.0             metapod_1.10.0           </span></span>
<span><span class="co">#&gt;  [19] reticulate_1.43.0         DBI_1.2.3                </span></span>
<span><span class="co">#&gt;  [21] RColorBrewer_1.1-3        lubridate_1.9.4          </span></span>
<span><span class="co">#&gt;  [23] abind_1.4-8               zlibbioc_1.48.2          </span></span>
<span><span class="co">#&gt;  [25] rvest_1.0.5               quadprog_1.5-8           </span></span>
<span><span class="co">#&gt;  [27] purrr_1.1.0               RCurl_1.98-1.17          </span></span>
<span><span class="co">#&gt;  [29] pracma_2.4.4              rappdirs_0.3.3           </span></span>
<span><span class="co">#&gt;  [31] multimode_1.5             circlize_0.4.16          </span></span>
<span><span class="co">#&gt;  [33] GenomeInfoDbData_1.2.11   irlba_2.3.5.1            </span></span>
<span><span class="co">#&gt;  [35] terra_1.8-60              units_0.8-7              </span></span>
<span><span class="co">#&gt;  [37] dqrng_0.4.1               pkgdown_2.1.3            </span></span>
<span><span class="co">#&gt;  [39] DelayedMatrixStats_1.24.0 codetools_0.2-20         </span></span>
<span><span class="co">#&gt;  [41] DelayedArray_0.28.0       scuttle_1.12.0           </span></span>
<span><span class="co">#&gt;  [43] xml2_1.4.0                tidyselect_1.2.1         </span></span>
<span><span class="co">#&gt;  [45] shape_1.4.6.1             farver_2.1.2             </span></span>
<span><span class="co">#&gt;  [47] ScaledMatrix_1.10.0       jsonlite_2.0.0           </span></span>
<span><span class="co">#&gt;  [49] BiocNeighbors_1.20.0      e1071_1.7-16             </span></span>
<span><span class="co">#&gt;  [51] ks_1.15.1                 Formula_1.2-5            </span></span>
<span><span class="co">#&gt;  [53] ggridges_0.5.7            systemfonts_1.2.3        </span></span>
<span><span class="co">#&gt;  [55] dbscan_1.2.3              tools_4.3.3              </span></span>
<span><span class="co">#&gt;  [57] progress_1.2.3            ragg_1.5.0               </span></span>
<span><span class="co">#&gt;  [59] Rcpp_1.1.0                glue_1.8.0               </span></span>
<span><span class="co">#&gt;  [61] mnormt_2.1.1              SparseArray_1.2.4        </span></span>
<span><span class="co">#&gt;  [63] xfun_0.53                 decoupleR_2.8.0          </span></span>
<span><span class="co">#&gt;  [65] dplyr_1.1.4               withr_3.0.2              </span></span>
<span><span class="co">#&gt;  [67] fastmap_1.2.0             bluster_1.12.0           </span></span>
<span><span class="co">#&gt;  [69] digest_0.6.37             rsvd_1.0.5               </span></span>
<span><span class="co">#&gt;  [71] timechange_0.3.0          R6_2.6.1                 </span></span>
<span><span class="co">#&gt;  [73] textshaping_1.0.1         colorspace_2.1-1         </span></span>
<span><span class="co">#&gt;  [75] dichromat_2.0-0.1         diptest_0.77-2           </span></span>
<span><span class="co">#&gt;  [77] tidyr_1.3.1               generics_0.1.4           </span></span>
<span><span class="co">#&gt;  [79] data.table_1.17.8         class_7.3-23             </span></span>
<span><span class="co">#&gt;  [81] prettyunits_1.2.0         httr_1.4.7               </span></span>
<span><span class="co">#&gt;  [83] htmlwidgets_1.6.4         S4Arrays_1.2.1           </span></span>
<span><span class="co">#&gt;  [85] pkgconfig_2.0.3           gtable_0.3.6             </span></span>
<span><span class="co">#&gt;  [87] S7_0.2.0                  XVector_0.42.0           </span></span>
<span><span class="co">#&gt;  [89] OmnipathR_3.10.1          htmltools_0.5.8.1        </span></span>
<span><span class="co">#&gt;  [91] carData_3.0-5             scales_1.4.0             </span></span>
<span><span class="co">#&gt;  [93] png_0.1-8                 corrplot_0.95            </span></span>
<span><span class="co">#&gt;  [95] scran_1.30.0              knitr_1.50               </span></span>
<span><span class="co">#&gt;  [97] tzdb_0.5.0                rjson_0.2.23             </span></span>
<span><span class="co">#&gt;  [99] checkmate_2.3.3           nlme_3.1-168             </span></span>
<span><span class="co">#&gt; [101] curl_7.0.0                proxy_0.4-27             </span></span>
<span><span class="co">#&gt; [103] cachem_1.1.0              GlobalOptions_0.1.2      </span></span>
<span><span class="co">#&gt; [105] stringr_1.5.1             testit_0.13              </span></span>
<span><span class="co">#&gt; [107] rootSolve_1.8.2.4         KernSmooth_2.23-26       </span></span>
<span><span class="co">#&gt; [109] parallel_4.3.3            RcppZiggurat_0.1.8       </span></span>
<span><span class="co">#&gt; [111] desc_1.4.3                pillar_1.11.1            </span></span>
<span><span class="co">#&gt; [113] grid_4.3.3                logger_0.4.1             </span></span>
<span><span class="co">#&gt; [115] vctrs_0.6.5               ggpubr_0.6.1             </span></span>
<span><span class="co">#&gt; [117] car_3.1-3                 BiocSingular_1.18.0      </span></span>
<span><span class="co">#&gt; [119] beachmat_2.18.1           cluster_2.1.8.1          </span></span>
<span><span class="co">#&gt; [121] evaluate_1.0.5            readr_2.1.5              </span></span>
<span><span class="co">#&gt; [123] magick_2.8.7              mvtnorm_1.3-3            </span></span>
<span><span class="co">#&gt; [125] cli_3.6.5                 locfit_1.5-9.12          </span></span>
<span><span class="co">#&gt; [127] compiler_4.3.3            rlang_1.1.6              </span></span>
<span><span class="co">#&gt; [129] crayon_1.5.3              ggsignif_0.6.4           </span></span>
<span><span class="co">#&gt; [131] labeling_0.4.3            mclust_6.1.1             </span></span>
<span><span class="co">#&gt; [133] classInt_0.4-11           fs_1.6.6                 </span></span>
<span><span class="co">#&gt; [135] stringi_1.8.7             psych_2.5.6              </span></span>
<span><span class="co">#&gt; [137] BiocParallel_1.36.0       Matrix_1.6-5             </span></span>
<span><span class="co">#&gt; [139] hms_1.1.4                 sparseMatrixStats_1.14.0 </span></span>
<span><span class="co">#&gt; [141] ggplot2_4.0.0             statmod_1.5.1            </span></span>
<span><span class="co">#&gt; [143] Rfast_2.1.0               igraph_2.1.4             </span></span>
<span><span class="co">#&gt; [145] broom_1.0.9               RcppParallel_5.1.9       </span></span>
<span><span class="co">#&gt; [147] bslib_0.9.0               Giotto_3.3.2             </span></span>
<span><span class="co">#&gt; [149] readxl_1.4.5</span></span></code></pre>
            </div>
          </div>
        </main>
        <aside class="col-md-3">
          <nav id="toc" aria-label="Table of contents">
            <h2>On this page</h2>
          </nav>
        </aside>
      </div>

      <footer>
        <div class="pkgdown-footer-left">
          <p>Developed by Constantin Zackl.</p>
        </div>

        <div class="pkgdown-footer-right">
          <p>
            Site built with
            <a href="https://pkgdown.r-lib.org/" class="external-link"
              >pkgdown</a
            >
            2.1.3.
          </p>
        </div>
      </footer>
    </div>
  </body>
</html>
