<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Data Input • spacedeconv</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Data Input">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/spacedeconv.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/spacedeconv_clustering.html">Clustering</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_data.html">Data Input</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_decoupleR.html">DecoupleR integration</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_example.html">Example</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_faq.html">FAQ</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_LR.html">Ligand-Receptor Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_utils.html">Workflow Utilities</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_visualization.html">Visualization</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Data Input</h1>
            
      

      <div class="d-none name"><code>spacedeconv_data.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>spacedeconv integrates first- and second-generation deconvolution
algorithms for transcriptome data. While first-generation tools
deconvolute based on internal precomputed signatures, second-generation
deconvolution tools compute cell type specific expression signatures
from annotated single-cell expression data. In the following
requirements and workflow details for both types of data are explained
in detail.</p>
</div>
<div class="section level2">
<h2 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>spacedeconv contains 4 different spatial datasets obtained with the
10XVisium platform. Additionally there are 4 matching scRNA-seq datasets
available which can be used to calculate custom expression signatures.
The data can be loaded like this:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://omnideconv.github.io/spacedeconv">spacedeconv</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; → Setting up spacedeconv environment..</span></span>
<span><span class="co">#&gt; → Using conda environment 'r-omnideconv'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_4"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reference-data">Reference Data<a class="anchor" aria-label="anchor" href="#reference-data"></a>
</h2>
<p>To build cell-type specific signatures for second-generation
deconvolution spacedeconv utilized annotated single-cell reference data.
Your single cell data needs to include cell-type annotations and can be
one of the following formats:</p>
<ul>
<li>
<a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html#1_Class_structure" class="external-link">SingleCellExperiment</a>
(recommended)</li>
<li><a href="https://satijalab.org/seurat/" class="external-link">Seurat</a></li>
</ul>
<p>Cell type information needs to be available. You can specify the
column containing the annotation with the <code>cell_type_col</code>
parameter. The same applies to batch ID information required for CARD
with the parameter <code>batch_id_col</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span><span class="va">single_cell_data_1</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"spatialdwls"</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># some methods require batch_id information as well</span></span>
<span><span class="va">sigature</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span><span class="va">single_cell_data_1</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"cell2location"</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  batch_id_col <span class="op">=</span> <span class="st">"orig.ident"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-data">Spatial Data<a class="anchor" aria-label="anchor" href="#spatial-data"></a>
</h2>
<p>Spatially resolved data needs to be available in the
SpatialExperiment format. For 10XVisium slides the data can be loaded
easily by providing the path to the directory create by <a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger" class="external-link">SpaceRanger</a>.
More information about data loading and manual object construction can
be found in the SpatialExperiment <a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html" class="external-link">Vignette</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">SpatialExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/read10xVisium.html" class="external-link">read10xVisium</a></span><span class="op">(</span><span class="va">path_to_directory</span><span class="op">)</span></span></code></pre></div>
<p>Spatial data available as a Seurat Object can be converted into a
SpatialExperiment:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/seurat_to_spatialexperiment.html">seurat_to_spatialexperiment</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<p>SpaceDeconv offers an additional function for convenient
normalization of SpatialExperiments. The normalization is saved in a new
assay, so make sure the correct data is used during deconvolution by
providing the desired assay with the parameters <code>assay_sc</code>
and <code>assay_sp</code>. As normalization method <code>cpm</code> and
<code>logcpm</code> are available.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">spatial_data_1</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[64ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 247 observations with umi count below threshold</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 247 observations with umi count below threshold <span style="color: #B2B2B2;">[193ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 13479 variables with all zero expression</span></span>
<span><span class="co">#&gt; Warning in spacedeconv::preprocess(spatial_data_1): There are 13 mitochondrial</span></span>
<span><span class="co">#&gt; genes present. Consider removing them.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 13479 variables with all zero expression <span style="color: #B2B2B2;">[820ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Warning: ENSEMBL identifiers detected in gene names</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BBBB;">ℹ</span> Consider using Gene Names for first-generation deconvolution tools</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BB00;">✔</span> Finished Preprocessing <span style="color: #B2B2B2;">[9ms]</span></span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spe</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[16ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Normalizing using cpm</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Finished normalization using cpm <span style="color: #B2B2B2;">[2.1s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please note the normalization is stored in an additional assay</span></span>
<span></span>
<span><span class="co"># make sure to use cpm assay in deconvolution step</span></span>
<span><span class="va">deconvolution</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span><span class="va">spe</span>, method <span class="op">=</span> <span class="st">"quantiseq"</span>, assay_sp <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[22ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Spatial </span></span>
<span><span class="co">#&gt; Assays: <span style="color: #0000BB;">"counts"</span> and <span style="color: #0000BB;">"cpm"</span></span></span>
<span><span class="co">#&gt; Genes: <span style="color: #0000BB;">23112</span></span></span>
<span><span class="co">#&gt; → without expression: <span style="color: #0000BB;">0</span> (0%)</span></span>
<span><span class="co">#&gt; Spots: <span style="color: #0000BB;">2467</span></span></span>
<span><span class="co">#&gt; Spots under tissue: <span style="color: #0000BB;">2467</span> (100%)</span></span>
<span><span class="co">#&gt; Median Genes Per Spot: <span style="color: #0000BB;">3329</span></span></span>
<span><span class="co">#&gt; → without expression: <span style="color: #0000BB;">0</span> (0%)</span></span>
<span><span class="co">#&gt; Umi count range: <span style="color: #0000BB;">500</span> - <span style="color: #0000BB;">58175</span></span></span>
<span><span class="co">#&gt; Spots with UMI count below 500: <span style="color: #0000BB;">0</span> (0%)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Rownames set</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Colnames set</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> deconvoluting</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; &gt;&gt;&gt; Running quantiseq</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running quanTIseq deconvolution module</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Removing 17 noisy genes</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Removing 15 genes with high expression in tumors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Signature genes found in data set: 134/138 (97.1%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Mixture deconvolution (method: lsei)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Deconvolution successful!</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> finished <span style="color: #B2B2B2;">[28.3s]</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="image-alignment">Image Alignment<a class="anchor" aria-label="anchor" href="#image-alignment"></a>
</h2>
<p>In case the background image is not aligned properly the
SpatialExperiment class offers convenient functions for rotation /
mirroring:</p>
<ul>
<li><code>rotateImg(image, degrees)</code></li>
<li><code>mirrorImg(image, axis) # 'h'/'v'</code></li>
</ul>
<p>More Information is available in the SpatialExperiment <a href="https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html#34_Image_transformations" class="external-link">Documentation</a></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
