<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Example • spacedeconv</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Example">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/spacedeconv.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/spacedeconv_clustering.html">Clustering</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_data.html">Data Input</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_example.html">Example</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_faq.html">FAQ</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_LR.html">Ligand-Receptor Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_utils.html">Workflow Utilities</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_visualization.html">Visualization</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv-decoupleR.html">DecoupleR integration</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Example</h1>
            
      

      <div class="d-none name"><code>spacedeconv_example.Rmd</code></div>
    </div>

    
    
<p>spacedeconv is a unified interface for the deconvolution of spatial
transcriptomics data. In total spacedeconv gives access to 31
deconvolution methods:</p>
<pre><code><span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:spacedeconv':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     normalize</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="co">#&gt;             RCTD        SPOTlight             CARD      spatialDWLS </span></span>
<span><span class="co">#&gt;           "rctd"      "spotlight"           "card"    "spatialdwls" </span></span>
<span><span class="co">#&gt;    cell2location        AutoGeneS       BayesPrism           Bisque </span></span>
<span><span class="co">#&gt;  "cell2location"      "autogenes"     "bayesprism"         "bisque" </span></span>
<span><span class="co">#&gt;          BSeq-sc       CIBERSORTx            CDSeq              CPM </span></span>
<span><span class="co">#&gt;         "bseqsc"     "cibersortx"          "cdseq"            "cpm" </span></span>
<span><span class="co">#&gt;             DWLS             MOMF            MuSiC           Scaden </span></span>
<span><span class="co">#&gt;           "dwls"           "momf"          "music"         "scaden" </span></span>
<span><span class="co">#&gt;             SCDC       MCPcounter             EPIC        quanTIseq </span></span>
<span><span class="co">#&gt;           "scdc"    "mcp_counter"           "epic"      "quantiseq" </span></span>
<span><span class="co">#&gt;            xCell        CIBERSORT CIBERSORT (abs.)            TIMER </span></span>
<span><span class="co">#&gt;          "xcell"      "cibersort"  "cibersort_abs"          "timer" </span></span>
<span><span class="co">#&gt;     ConsensusTME             ABIS         ESTIMATE      mMCPcounter </span></span>
<span><span class="co">#&gt;  "consensus_tme"           "abis"       "estimate"   "mmcp_counter" </span></span>
<span><span class="co">#&gt;        seqImmuCC              DCQ             BASE </span></span>
<span><span class="co">#&gt;      "seqimmucc"            "dcq"           "base"</span></span></code></pre>
<div class="section level2">
<h2 id="sample-data">Sample Data<a class="anchor" aria-label="anchor" href="#sample-data"></a>
</h2>
<p>For this tutorial we will use one of spacedeconvs sample dataset and
the deconvolution algorithm “SCDC”.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_3"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing-and-normalization">Preprocessing and Normalization<a class="anchor" aria-label="anchor" href="#preprocessing-and-normalization"></a>
</h2>
<p>To get a first insight into the data we type the dataset name into
the terminal:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 29733 7986 </span></span>
<span><span class="co">#&gt; metadata(1): Samples</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(29733): RP11-34P13.7 FO538757.3 ... KRTAP9-2 IGLVIV-66-1</span></span>
<span><span class="co">#&gt; rowData names(1): ID</span></span>
<span><span class="co">#&gt; colnames(7986): CID44971_AAGCCGCCACGCATCG CID44971_AAGGAGCTCAACACAC ...</span></span>
<span><span class="co">#&gt;   CID44971_TTTCCTCCAAGCCATT CID44971_TTTGGTTGTATCACCA</span></span>
<span><span class="co">#&gt; colData names(10): Sample Barcode ... celltype_minor celltype_major</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>We can see that the single-cell data is available as a <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a>.
In total this dataset includes expression values from 29733 genes and
7986 cells. It can further be seen that cell type information is
available in <code>colData</code> where cell related annotation is
stored.</p>
<p>As the next step we normalize the data as <code>cpm</code> (“Counts
Per Million”). The normalization is stored as an additional assay in the
single-cell object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single_cell_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[52ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 8 observations with umi count below threshold</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 8 observations with umi count below threshold <span style="color: #B2B2B2;">[1.5s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 5862 variables with all zero expression</span></span>
<span><span class="co">#&gt; Warning in spacedeconv::preprocess(single_cell_data_3): There are 13</span></span>
<span><span class="co">#&gt; mitochondrial genes present. Consider removing them.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 5862 variables with all zero expression <span style="color: #B2B2B2;">[811ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Warning: ENSEMBL identifiers detected in gene names</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BBBB;">ℹ</span> Consider using Gene Names for first-generation deconvolution tools</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BB00;">✔</span> Finished Preprocessing <span style="color: #B2B2B2;">[7ms]</span></span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">spatial_data_3</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[25ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 137 observations with umi count below threshold</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 137 observations with umi count below threshold <span style="color: #B2B2B2;">[99ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing 13049 variables with all zero expression</span></span>
<span><span class="co">#&gt; Warning in spacedeconv::preprocess(spatial_data_3): There are 13 mitochondrial</span></span>
<span><span class="co">#&gt; genes present. Consider removing them.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removed 13049 variables with all zero expression <span style="color: #B2B2B2;">[113ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Warning: ENSEMBL identifiers detected in gene names</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BBBB;">ℹ</span> Consider using Gene Names for first-generation deconvolution tools</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Checking for ENSEMBL Identifiers<span style="color: #00BB00;">✔</span> Finished Preprocessing <span style="color: #B2B2B2;">[7ms]</span></span></span>
<span></span>
<span><span class="va">single_cell_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">single_cell_data_3</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[11ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Normalizing using cpm</span></span>
<span><span class="co">#&gt; Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size</span></span>
<span><span class="co">#&gt; 1.4 GiB</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Finished normalization using cpm <span style="color: #B2B2B2;">[4.9s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please note the normalization is stored in an additional assay</span></span>
<span><span class="va">spatial_data_3</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spatial_data_3</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── spacedeconv ─────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> testing parameter</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> parameter OK <span style="color: #B2B2B2;">[11ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Normalizing using cpm</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Finished normalization using cpm <span style="color: #B2B2B2;">[483ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please note the normalization is stored in an additional assay</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="build-model">Build Model<a class="anchor" aria-label="anchor" href="#build-model"></a>
</h2>
<p>To build a model we need to provide a single cell expression object
annotated with cell type information. Let’s have a look into the object
to see which annotation is available.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Sample"          "Barcode"         "orig.ident"      "nCount_RNA"     </span></span>
<span><span class="co">#&gt;  [5] "nFeature_RNA"    "percent.mito"    "subtype"         "celltype_subset"</span></span>
<span><span class="co">#&gt;  [9] "celltype_minor"  "celltype_major"</span></span></code></pre></div>
<p>There are three different resolutions of cell type annotation
available. For this tutorial we will choose
<code>"celltype_major"</code> as cell type annotation which includes 9
different cell-types.</p>
<p>Next we can calculate a signature using the annotated single-data and
the deconvolution tool “spatialDWLS”.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signature</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"dwls"</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span>, dwls_method <span class="op">=</span> <span class="st">"mast_optimized"</span>, ncores <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The signature contains expression values for all celltypes and a
subset of distinctly expressed genes. The following example shows the
first 10 rows of the signature calculated in the previous step.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">signature</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="11%">
<col width="6%">
<col width="6%">
<col width="7%">
<col width="12%">
<col width="7%">
<col width="7%">
<col width="16%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">Endothelial</th>
<th align="right">CAFs</th>
<th align="right">PVL</th>
<th align="right">B-cells</th>
<th align="right">Plasmablasts</th>
<th align="right">T-cells</th>
<th align="right">Myeloid</th>
<th align="right">Cancer Epithelial</th>
<th align="right">Normal Epithelial</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ACKR1</td>
<td align="right">15.1019</td>
<td align="right">0.0086</td>
<td align="right">0.0110</td>
<td align="right">0.0081</td>
<td align="right">0.0208</td>
<td align="right">0.0082</td>
<td align="right">0.0118</td>
<td align="right">0.0089</td>
<td align="right">0.0068</td>
</tr>
<tr class="even">
<td align="left">PLVAP</td>
<td align="right">13.4028</td>
<td align="right">0.0172</td>
<td align="right">0.0110</td>
<td align="right">0.0081</td>
<td align="right">0.1250</td>
<td align="right">0.0112</td>
<td align="right">0.1853</td>
<td align="right">0.0168</td>
<td align="right">0.0068</td>
</tr>
<tr class="odd">
<td align="left">RBP7</td>
<td align="right">10.5972</td>
<td align="right">0.0103</td>
<td align="right">0.0110</td>
<td align="right">0.0027</td>
<td align="right">0.8125</td>
<td align="right">0.0055</td>
<td align="right">0.1103</td>
<td align="right">0.0213</td>
<td align="right">0.3038</td>
</tr>
<tr class="even">
<td align="left">RAMP2</td>
<td align="right">9.1157</td>
<td align="right">0.2646</td>
<td align="right">0.0440</td>
<td align="right">0.0027</td>
<td align="right">0.1250</td>
<td align="right">0.0048</td>
<td align="right">0.0176</td>
<td align="right">0.0492</td>
<td align="right">0.0817</td>
</tr>
<tr class="odd">
<td align="left">FABP4</td>
<td align="right">30.1806</td>
<td align="right">0.1134</td>
<td align="right">9.7582</td>
<td align="right">0.0434</td>
<td align="right">1.5000</td>
<td align="right">0.0186</td>
<td align="right">2.3059</td>
<td align="right">0.2864</td>
<td align="right">0.0272</td>
</tr>
<tr class="even">
<td align="left">AQP1</td>
<td align="right">7.3241</td>
<td align="right">0.2354</td>
<td align="right">0.0549</td>
<td align="right">0.0054</td>
<td align="right">0.1250</td>
<td align="right">0.0057</td>
<td align="right">0.0118</td>
<td align="right">0.0190</td>
<td align="right">0.0518</td>
</tr>
<tr class="odd">
<td align="left">VWF</td>
<td align="right">5.4907</td>
<td align="right">0.0017</td>
<td align="right">0.0000</td>
<td align="right">0.0027</td>
<td align="right">0.0208</td>
<td align="right">0.0021</td>
<td align="right">0.0059</td>
<td align="right">0.0089</td>
<td align="right">0.0041</td>
</tr>
<tr class="even">
<td align="left">SDPR</td>
<td align="right">5.7500</td>
<td align="right">0.0052</td>
<td align="right">0.3626</td>
<td align="right">0.0000</td>
<td align="right">0.0833</td>
<td align="right">0.0032</td>
<td align="right">0.0206</td>
<td align="right">0.0268</td>
<td align="right">0.0463</td>
</tr>
<tr class="odd">
<td align="left">CLEC14A</td>
<td align="right">4.8843</td>
<td align="right">0.1340</td>
<td align="right">0.0440</td>
<td align="right">0.0108</td>
<td align="right">0.0208</td>
<td align="right">0.0021</td>
<td align="right">0.0044</td>
<td align="right">0.0078</td>
<td align="right">0.0041</td>
</tr>
<tr class="even">
<td align="left">EMCN</td>
<td align="right">3.8287</td>
<td align="right">0.0155</td>
<td align="right">0.0330</td>
<td align="right">0.0027</td>
<td align="right">0.0000</td>
<td align="right">0.0018</td>
<td align="right">0.0029</td>
<td align="right">0.0045</td>
<td align="right">0.0027</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="deconvolution">Deconvolution<a class="anchor" aria-label="anchor" href="#deconvolution"></a>
</h2>
<p>To deconvolute the Visium slide we use the signature we just
calculated. Depending on the selected deconvolution tool further
parameters need to be passed, in this case <code>cell_type_col</code>
and <code>batch_id_col</code> as well as the scRNA-seq reference
dataset.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deconv</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  spatial_obj <span class="op">=</span> <span class="va">spatial_data_3</span>,</span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">single_cell_data_3</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"dwls"</span>,</span>
<span>  signature <span class="op">=</span> <span class="va">signature</span>,</span>
<span>  assay_sp <span class="op">=</span> <span class="st">"cpm"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>The deconvolution results are stored in the spatial object to
simplify visualizations. spacedeconv offers 6 visualization functions
with different scopes, further explained in the visualization <a href="spacedeconv_visualization.html">vignette</a>.</p>
<p>We can see the available deconvolution results by using
<code>available_results(deconv)</code> or by accessing the objects
<code>colData</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/available_results.html">available_results</a></span><span class="op">(</span><span class="va">deconv</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "dwls_B.cells"           "dwls_CAFs"              "dwls_Cancer.Epithelial"</span></span>
<span><span class="co">#&gt; [4] "dwls_Endothelial"       "dwls_Myeloid"           "dwls_Normal.Epithelial"</span></span>
<span><span class="co">#&gt; [7] "dwls_Plasmablasts"      "dwls_PVL"               "dwls_T.cells"</span></span></code></pre></div>
<p>For each Spot the estimated cell fraction is available as object
annotation and can be used for visualizations.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot all available results: provide the tool as parameter</span></span>
<span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  cell_type <span class="op">=</span> <span class="st">"dwls_B.cells"</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  smooth <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  title_size <span class="op">=</span> <span class="fl">12</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_example_files/figure-html/visualization-1.png" width="1152"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] SpatialExperiment_1.14.0    SingleCellExperiment_1.26.0</span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.34.0 Biobase_2.64.0             </span></span>
<span><span class="co">#&gt;  [5] GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt;  [7] IRanges_2.38.1              S4Vectors_0.42.1           </span></span>
<span><span class="co">#&gt;  [9] BiocGenerics_0.50.0         MatrixGenerics_1.16.0      </span></span>
<span><span class="co">#&gt; [11] matrixStats_1.4.1           spacedeconv_0.1.1          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] later_1.3.2             tibble_3.2.1            R.oo_1.26.0            </span></span>
<span><span class="co">#&gt;   [4] cellranger_1.1.0        lifecycle_1.0.4         sf_1.0-17              </span></span>
<span><span class="co">#&gt;   [7] rstatix_0.7.2           edgeR_4.2.1             lattice_0.22-6         </span></span>
<span><span class="co">#&gt;  [10] backports_1.5.0         magrittr_2.0.3          limma_3.60.6           </span></span>
<span><span class="co">#&gt;  [13] plotly_4.10.4           sass_0.4.9              rmarkdown_2.28         </span></span>
<span><span class="co">#&gt;  [16] jquerylib_0.1.4         yaml_2.3.10             remotes_2.5.0          </span></span>
<span><span class="co">#&gt;  [19] sp_2.1-4                reticulate_1.39.0       cowplot_1.1.3          </span></span>
<span><span class="co">#&gt;  [22] DBI_1.2.3               RColorBrewer_1.1-3      lubridate_1.9.3        </span></span>
<span><span class="co">#&gt;  [25] abind_1.4-8             zlibbioc_1.50.0         rvest_1.0.4            </span></span>
<span><span class="co">#&gt;  [28] purrr_1.0.2             R.utils_2.12.3          pracma_2.4.4           </span></span>
<span><span class="co">#&gt;  [31] rappdirs_0.3.3          multimode_1.5           circlize_0.4.16        </span></span>
<span><span class="co">#&gt;  [34] GenomeInfoDbData_1.2.12 ggrepel_0.9.6           terra_1.7-78           </span></span>
<span><span class="co">#&gt;  [37] units_0.8-5             parallelly_1.38.0       pkgdown_2.1.1          </span></span>
<span><span class="co">#&gt;  [40] colorRamp2_0.1.0        codetools_0.2-20        DelayedArray_0.30.1    </span></span>
<span><span class="co">#&gt;  [43] xml2_1.3.6              tidyselect_1.2.1        shape_1.4.6.1          </span></span>
<span><span class="co">#&gt;  [46] farver_2.1.2            UCSC.utils_1.0.0        jsonlite_1.8.9         </span></span>
<span><span class="co">#&gt;  [49] e1071_1.7-16            ks_1.14.3               Formula_1.2-5          </span></span>
<span><span class="co">#&gt;  [52] ggridges_0.5.6          systemfonts_1.1.0       dbscan_1.2-0           </span></span>
<span><span class="co">#&gt;  [55] tools_4.4.1             progress_1.2.3          GiottoClass_0.3.5      </span></span>
<span><span class="co">#&gt;  [58] ragg_1.3.3              Rcpp_1.0.13             glue_1.8.0             </span></span>
<span><span class="co">#&gt;  [61] mnormt_2.1.1            SparseArray_1.4.8       xfun_0.48              </span></span>
<span><span class="co">#&gt;  [64] decoupleR_2.10.0        dplyr_1.1.4             withr_3.0.1            </span></span>
<span><span class="co">#&gt;  [67] fastmap_1.2.0           fansi_1.0.6             digest_0.6.37          </span></span>
<span><span class="co">#&gt;  [70] timechange_0.3.0        R6_2.5.1                textshaping_0.4.0      </span></span>
<span><span class="co">#&gt;  [73] colorspace_2.1-1        scattermore_1.2         gtools_3.9.5           </span></span>
<span><span class="co">#&gt;  [76] diptest_0.77-1          R.methodsS3_1.8.2       utf8_1.2.4             </span></span>
<span><span class="co">#&gt;  [79] tidyr_1.3.1             generics_0.1.3          data.table_1.16.2      </span></span>
<span><span class="co">#&gt;  [82] class_7.3-22            prettyunits_1.2.0       httr_1.4.7             </span></span>
<span><span class="co">#&gt;  [85] htmlwidgets_1.6.4       S4Arrays_1.4.1          pkgconfig_2.0.3        </span></span>
<span><span class="co">#&gt;  [88] gtable_0.3.5            GiottoVisuals_0.2.4     XVector_0.44.0         </span></span>
<span><span class="co">#&gt;  [91] OmnipathR_3.12.4        htmltools_0.5.8.1       carData_3.0-5          </span></span>
<span><span class="co">#&gt;  [94] scales_1.3.0            GiottoUtils_0.1.12      png_0.1-8              </span></span>
<span><span class="co">#&gt;  [97] corrplot_0.94           knitr_1.48              tzdb_0.4.0             </span></span>
<span><span class="co">#&gt; [100] reshape2_1.4.4          rjson_0.2.23            checkmate_2.3.2        </span></span>
<span><span class="co">#&gt; [103] nlme_3.1-164            curl_5.2.3              proxy_0.4-27           </span></span>
<span><span class="co">#&gt; [106] cachem_1.1.0            GlobalOptions_0.1.2     stringr_1.5.1          </span></span>
<span><span class="co">#&gt; [109] testit_0.13             rootSolve_1.8.2.4       KernSmooth_2.23-24     </span></span>
<span><span class="co">#&gt; [112] parallel_4.4.1          desc_1.4.3              pillar_1.9.0           </span></span>
<span><span class="co">#&gt; [115] grid_4.4.1              logger_0.3.0            vctrs_0.6.5            </span></span>
<span><span class="co">#&gt; [118] ggpubr_0.6.0            car_3.1-3               evaluate_1.0.1         </span></span>
<span><span class="co">#&gt; [121] readr_2.1.5             magick_2.8.5            mvtnorm_1.3-1          </span></span>
<span><span class="co">#&gt; [124] cli_3.6.3               locfit_1.5-9.10         compiler_4.4.1         </span></span>
<span><span class="co">#&gt; [127] rlang_1.1.4             crayon_1.5.3            ggsignif_0.6.4         </span></span>
<span><span class="co">#&gt; [130] labeling_0.4.3          mclust_6.1.1            classInt_0.4-10        </span></span>
<span><span class="co">#&gt; [133] plyr_1.8.9              fs_1.6.4                stringi_1.8.4          </span></span>
<span><span class="co">#&gt; [136] psych_2.4.6.26          viridisLite_0.4.2       deldir_2.0-4           </span></span>
<span><span class="co">#&gt; [139] BiocParallel_1.38.0     munsell_0.5.1           lazyeval_0.2.2         </span></span>
<span><span class="co">#&gt; [142] Matrix_1.7-0            hms_1.1.3               ggplot2_3.5.1          </span></span>
<span><span class="co">#&gt; [145] statmod_1.5.0           highr_0.11              igraph_2.0.3           </span></span>
<span><span class="co">#&gt; [148] broom_1.0.7             bslib_0.8.0             Giotto_4.1.0           </span></span>
<span><span class="co">#&gt; [151] readxl_1.4.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
