<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualization • spacedeconv</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Visualization">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/spacedeconv.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/spacedeconv_clustering.html">Clustering</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_data.html">Data Input</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_example.html">Example</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_faq.html">FAQ</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_LR.html">Ligand-Receptor Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_utils.html">Workflow Utilities</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv_visualization.html">Visualization</a></li>
    <li><a class="dropdown-item" href="../articles/spacedeconv-decoupleR.html">DecoupleR integration</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Visualization</h1>
            
      

      <div class="d-none name"><code>spacedeconv_visualization.Rmd</code></div>
    </div>

    
    
<p>To introduce spacedeconvs visualization functions we will utilize a
deconvolution result obtained from the sample data provided and the
deconvolution method <code>DWLS</code>. First we compute a reference
signature based on the available scRNA-seq data with cell-type
annotation. Then we quantify cell-type fractions in the 10X Visium slide
using the computed reference signature.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://omnideconv.github.io/spacedeconv">spacedeconv</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spatial_data_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_3"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">single_cell_data_3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess.html">preprocess</a></span><span class="op">(</span><span class="va">spatial_data_3</span><span class="op">)</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">spe</span>, method <span class="op">=</span> <span class="st">"cpm"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">singnature</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  single_cell_obj <span class="op">=</span> <span class="va">sce</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"dwls"</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">deconv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  <span class="va">spe</span>,</span>
<span>  signature <span class="op">=</span> <span class="va">signature</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"dwls"</span>,</span>
<span>  assay_sp <span class="op">=</span> <span class="st">"cpm"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="available-visualizations">Available Visualizations<a class="anchor" aria-label="anchor" href="#available-visualizations"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<code><a href="../reference/plot_celltype.html">plot_celltype()</a></code> visualizes the spatial image with color
coded cell type fractions for one cell type.</li>
<li>
<code><a href="../reference/plot_gene.html">plot_gene()</a></code> plot spatial gene expression.</li>
<li>
<code><a href="../reference/plot_umi_count.html">plot_umi_count()</a></code> Plots the number of sequenced reads
per spot.</li>
<li>
<code><a href="../reference/plot_most_abundant.html">plot_most_abundant()</a></code> Render a plot containing the most
abundant cell-type for each spot.</li>
<li>
<code><a href="../reference/plot_comparison.html">plot_comparison()</a></code> Plot comparison of two cell-type
fractions.</li>
</ol>
<div class="section level3">
<h3 id="plot_celltype">1. <code>plot_celltype()</code><a class="anchor" aria-label="anchor" href="#plot_celltype"></a>
</h3>
<p>Plot any spot annotation with a continuous or discrete scale. The
spot annotation needs to be of <code>colData(spe)</code>, so manual
annotation can be added to the SpatialExperiment object for
visualization.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">deconv</span>, cell_type <span class="op">=</span> <span class="st">"dwls_B.cells"</span>, density <span class="op">=</span> <span class="cn">F</span>, smooth <span class="op">=</span> <span class="cn">T</span>, title <span class="op">=</span> <span class="st">"B cells"</span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_visualization_files/figure-html/unnamed-chunk-2-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="plot_gene">2. <code>plot_gene()</code><a class="anchor" aria-label="anchor" href="#plot_gene"></a>
</h3>
<p>Plot gene expression on a spatial level. It may make sense to smooth
the plot to simplify the detection of expression patterns. You can
further select the assay using the <code>assay</code> parameter.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_gene.html">plot_gene</a></span><span class="op">(</span><span class="va">deconv</span>, gene <span class="op">=</span> <span class="st">"CD19"</span>, density <span class="op">=</span> <span class="cn">F</span>, smooth <span class="op">=</span> <span class="cn">T</span>, title <span class="op">=</span> <span class="st">"CD19 expression"</span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_visualization_files/figure-html/unnamed-chunk-3-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="plot_umi_count">3. <code>plot_umi_count()</code><a class="anchor" aria-label="anchor" href="#plot_umi_count"></a>
</h3>
<p>This plot shows the number of detected UMIs for each spot. We
recommend rendering this plot with <code>transform_scale = "sqrt"</code>
due to the large range of UMI count values.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_umi_count.html">plot_umi_count</a></span><span class="op">(</span><span class="va">deconv</span>, transform_scale <span class="op">=</span> <span class="st">"sqrt"</span>, density <span class="op">=</span> <span class="cn">F</span>, smooth <span class="op">=</span> <span class="cn">T</span>, title_size <span class="op">=</span> <span class="st">"UMI count"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in numnotnull("fontsize"): NAs introduced by coercion</span></span></code></pre>
<p><img src="spacedeconv_visualization_files/figure-html/plotUMIcount-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="plot_most_abundant">4. <code>plot_most_abundant()</code><a class="anchor" aria-label="anchor" href="#plot_most_abundant"></a>
</h3>
<p>This plots displays the most abundant cell-type for each spot. You
can specify which cells to plot by either one of the following:</p>
<ul>
<li>
<code>cell_type</code> vector of celltypes to plot</li>
<li>
<code>method</code> plotting all cell types of the provided
method</li>
<li>
<code>remove</code> vector of celltypes to be removed from the
plot</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_most_abundant.html">plot_most_abundant</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"dwls"</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  title_size <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  legend_size <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  remove <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dwls_Cancer.Epithelial"</span>, <span class="st">"dwls_CAFs"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_visualization_files/figure-html/plotMostAbundant-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="plot_comparison">5. <code>plot_comparison()</code><a class="anchor" aria-label="anchor" href="#plot_comparison"></a>
</h3>
<p>Use this function to plot the ratio of deconvolution results from two
cell-types.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_comparison.html">plot_comparison</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  cell_type_1 <span class="op">=</span> <span class="st">"dwls_Cancer.Epithelial"</span>, <span class="co"># red</span></span>
<span>  cell_type_2 <span class="op">=</span> <span class="st">"dwls_Normal.Epithelial"</span>, <span class="co"># blue</span></span>
<span>  palette <span class="op">=</span> <span class="st">"Blue-Red"</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  smooth <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Cancer vs. Normal Epithelial"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Negative Values in data, subtracting minimum to shift distribution to &gt; 0"</span></span></code></pre>
<p><img src="spacedeconv_visualization_files/figure-html/plotComparison-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="color-palette">Color Palette<a class="anchor" aria-label="anchor" href="#color-palette"></a>
</h2>
<p>All palettes from the <a href="https://colorspace.r-forge.r-project.org/" class="external-link">colorspace</a> R
Package can be used.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colorspace</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/colorspace/man/hcl_palettes.html" class="external-link">hcl_palettes</a></span><span class="op">(</span>plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_visualization_files/figure-html/availablePalettes-1.png" width="960"></p>
</div>
<div class="section level2">
<h2 id="further-plot-adjustments">Further plot adjustments<a class="anchor" aria-label="anchor" href="#further-plot-adjustments"></a>
</h2>
<div class="section level4">
<h4 id="image-alignment-offset">Image Alignment Offset<a class="anchor" aria-label="anchor" href="#image-alignment-offset"></a>
</h4>
<p>spacedeconvs Visualization function is designed to work with data by
SpaceRanger &gt;= V2.0. Since this Version the image is rotated by
default that the hourglass fiducial is in the upper left corner.
Previous SpaceRanger results can be rotated differently. The rotation
additionally reflects in the angle of the spots on the slide.
Uncorrectly rotated images result in hexagons being rotated by 30
degrees. To compensate for this you can use the
<code>offset_rotation</code> parameter to correct the hexagon alignment.
This is only necessary for Visium slides where the hourglass fiducial is
in the bottom left or upper right corner.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_umi_count.html">plot_umi_count</a></span><span class="op">(</span><span class="va">deconv</span>, offset_rotation <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="co"># rotate hexagons</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="transform-scale">Transform Scale<a class="anchor" aria-label="anchor" href="#transform-scale"></a>
</h4>
<p>With the <code>transform_scale</code> parameter the colorspace scale
can be modified. Available options are: “ln”, “log10”, “log2” and
“sqrt”. Scaling the color range differently can aid with interpreting
the plot. Please have in mind that the plot does not show valid
deconvolution results anymore and should be handled with caution.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_umi_count.html">plot_umi_count</a></span><span class="op">(</span><span class="va">deconv</span>, transform_scale <span class="op">=</span> <span class="st">"sqrt"</span>, density <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_visualization_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="smooth">Smooth<a class="anchor" aria-label="anchor" href="#smooth"></a>
</h4>
<p>With this parameter the expression values can be smoothed to simplify
pattern recognition. The smoother utilizes a linear kernel which size is
calculated by multiplying the spot distance with the smoothing factor.
It has to be mentioned that this operation changes the deconvolution
result in the final plots by appling the kernel, so they should be
interpreted as a helpful visualization option and not a deconvolution
result.</p>
<ul>
<li>
<code>smooth=T</code> enable smoothing</li>
<li>
<code>smoothing_factor</code> choose kernel size (factor of spot
distance)</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  cell_type <span class="op">=</span> <span class="st">"dwls_B.cells"</span>,</span>
<span>  smooth <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  smoothing_factor <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_visualization_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="density-distribution">Density Distribution<a class="anchor" aria-label="anchor" href="#density-distribution"></a>
</h4>
<p>You can add a density distribution by setting
<code>density = TRUE</code>. The red line corresponds to the mean of the
distribution.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  cell_type <span class="op">=</span> <span class="st">"dwls_B.cells"</span>,</span>
<span>  smooth <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  title_size <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  legend_size <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"B cells smoothed"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="spacedeconv_visualization_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="save-plots">Save Plots<a class="anchor" aria-label="anchor" href="#save-plots"></a>
</h4>
<p>You can save a plot by setting <code>save=TRUE</code>. Specify a file
path with the <code>path</code> parameter. If no path is provided your
plots will be stored at <code>~/spacedeconvResults/</code>. To change
the size of the rendered plot use <code>png_width</code> and
<code>png_height</code> to set the plot size in pixels. Plots are saved
as a png.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">spacedeconv</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_celltype.html">plot_celltype</a></span><span class="op">(</span><span class="va">deconv</span>,</span>
<span>  cell_type <span class="op">=</span> <span class="st">"quantiseq_B.cell"</span>,</span>
<span>  smooth <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  density <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  save <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  path <span class="op">=</span> <span class="st">"~/project/results"</span>,</span>
<span>  png_width <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  png_height <span class="op">=</span> <span class="fl">750</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="aggregate-cell-types">Aggregate cell types<a class="anchor" aria-label="anchor" href="#aggregate-cell-types"></a>
</h4>
<p>Aggregate cell types using the <code>aggregate</code> function.
Internall the deconvolution results are summed up and get a new name.
The aggregation can be visualized with all available plotting
functions.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">spe</span>, <span class="va">cell_type_1</span>, <span class="va">cell_type_2</span>, <span class="va">newName</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="additional-parameters">Additional Parameters<a class="anchor" aria-label="anchor" href="#additional-parameters"></a>
</h2>
<ul>
<li>
<code>show_image</code> logical, show or remove the spatial
image</li>
<li>
<code>spot_size</code> integer, increase (&gt;1) or decrease (&lt;1)
the hexagon size</li>
<li>
<code>limits</code> vector containing upper and lower limits for the
color scale</li>
<li>
<code>palette_type</code> “discrete”, “sequential” or “divergent”,
how to scale the color´</li>
<li>
<code>reverse_palette</code> reverse color palette</li>
<li>
<code>font_size</code> font size of legend</li>
<li>
<code>title_size</code> font size of title</li>
<li>
<code>legend_size</code> legend</li>
<li>
<code>title</code> set a custom title</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
