<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="spacedeconv">
<title>Metacell • spacedeconv</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Metacell">
<meta property="og:description" content="spacedeconv">
<meta property="og:image" content="https://omnideconv.github.io/spacedeconv/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">spacedeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.0000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/spacedeconv.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/spacedeconv_colocalization.html">Colocalization</a>
    <a class="dropdown-item" href="../articles/spacedeconv_data.html">Data Input</a>
    <a class="dropdown-item" href="../articles/spacedeconv_example.html">Example</a>
    <a class="dropdown-item" href="../articles/spacedeconv_faq.html">FAQ</a>
    <a class="dropdown-item" href="../articles/spacedeconv_metacell.html">Metacell</a>
    <a class="dropdown-item" href="../articles/spacedeconv_visualization.html">spacedeconv Visualization</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Metacell</h1>
            
      
      
      <div class="d-none name"><code>spacedeconv_metacell.Rmd</code></div>
    </div>

    
    
<p>Building cell-type specific expression signatures from annotated
single-cell data can be time comsuming. To reduce resource requirements
spacedeconv contains access to metacell2. The software is able to reduce
the dataset by computing robust “metacells” as a mixture of similar
cells from the originating single cell dataset. This approach can
improve the data input size by the factor 100.</p>
<p>In the following the metacell functions of spacedeconv are outlined
and integrated in a workflow.</p>
<p>We use spacedeconvs sample data for this analysis. Since Metacell is
python based we need to convert the SingleCellExperiment to an AnnData
object. The original dataset contains 5789 cells.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">spacedeconv</span><span class="op">)</span></span>
<span><span class="co">#&gt; Configuring package 'spacedeconv': please wait ...</span></span>
<span><span class="co">#&gt; Done!</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:spacedeconv':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     normalize</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"single_cell_data_1"</span><span class="op">)</span></span>
<span><span class="va">single_cell_data_1</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 29733 5789 </span></span>
<span><span class="co">#&gt; metadata(1): Samples</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(29733): RP11-34P13.7 FO538757.3 ... KRTAP9-2 IGLVIV-66-1</span></span>
<span><span class="co">#&gt; rowData names(1): ID</span></span>
<span><span class="co">#&gt; colnames(5789): CID4290A_AAACGGGAGACTGGGT CID4290A_AAAGTAGAGCGAAGGG ...</span></span>
<span><span class="co">#&gt;   CID4290A_TGCGTGGGTAGTAGTA CID4290A_TTTCCTCAGGCAGGTT</span></span>
<span><span class="co">#&gt; colData names(10): Sample Barcode ... celltype_minor celltype_major</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="va">ad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spe_to_ad.html">spe_to_ad</a></span><span class="op">(</span><span class="va">single_cell_data_1</span><span class="op">)</span> <span class="co"># convert to anndata</span></span></code></pre></div>
<p>The main workflow consists of 2 mandatory and 2 optional
functions:</p>
<ul>
<li>Filter Dataset</li>
<li>(optional) Compute forbidden genes and modules</li>
<li>(optional) Extract forbidden genes from modules</li>
<li>Compute Metacells</li>
</ul>
<div class="section level3">
<h3 id="clean-genes-and-cells">Clean Genes and Cells<a class="anchor" aria-label="anchor" href="#clean-genes-and-cells"></a>
</h3>
<p>The first step filters the dataset to remove low quality genes and
cells. It is possible to manually remove genes from the dataset and set
specific UMI cutoffs. For more instructions please view the functions
documentation.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clean_genes_and_cells.html">clean_genes_and_cells</a></span><span class="op">(</span><span class="va">ad</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── metacell ────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Cleaning genes and cells</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Cleaned genes and cells <span style="color: #B2B2B2;">[21.3s]</span></span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="optional-compute-forbidden-genes">(optional) compute forbidden genes<a class="anchor" aria-label="anchor" href="#optional-compute-forbidden-genes"></a>
</h3>
<p>This function finds genes in the dataset which should not be included
in the metacells and are calculated from hardcoded gene patterns. The
function returns a list of genes which can be used as input for the
metacell computation step. In addition gene modules are calculated and
stored in the single cell object which is used in the third step of this
workflow.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">suspect_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_forbidden_genes.html">compute_forbidden_genes</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── metacell ────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Computing forbidden genes</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Computed forbidden genes <span style="color: #B2B2B2;">[1.3s]</span></span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="optional-extract-forbidden-genes-from-gene-modules">(optional) extract forbidden genes from gene modules<a class="anchor" aria-label="anchor" href="#optional-extract-forbidden-genes-from-gene-modules"></a>
</h3>
<p>In this functions the genes from unsuited gene modules are extracted.
You just have to provide a list of unsuited gene modules and the
function returns an improved list of forbidden genes which can be used
as input for the last step in this workflow.</p>
<div class="section level4">
<h4 id="todo">TODO<a class="anchor" aria-label="anchor" href="#todo"></a>
</h4>
<p>here will be plots and instructions.</p>
</div>
</div>
<div class="section level3">
<h3 id="compute-metacells">Compute Metacells<a class="anchor" aria-label="anchor" href="#compute-metacells"></a>
</h3>
<p>This function uses the filtered single cell data and optional
forbidden genes to calculate metacells. Since metacells don’t have a
cell type annotation we reannotate the metacells based on the original
single cell data using the cell type column name you have to provide. It
is further possible to select an AbundanceScore to further subset the
metacells. The Abundance Score quantifies the purity of a metacells,
namely the percentage of the “most Abundant cell” in the metacell
compared to all cells. Not every cell merged to a metacell is of the
same cell type in the original dataset. Using the Abundance Score we
only keep metacells with more than 90% purity but other values can be
used as well.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metacells</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_metacells.html">compute_metacells</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="va">suspect_genes</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"celltype_major"</span>,</span>
<span>  abundance_score <span class="op">=</span> <span class="fl">0.9</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ── metacell ────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Computing metacells</span></span>
<span><span class="co">#&gt; Removing 1 metacell with abundance score under 0.9</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> computed metacells <span style="color: #B2B2B2;">[2.8s]</span></span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-result">The result<a class="anchor" aria-label="anchor" href="#the-result"></a>
</h2>
<p>The Input dataset was reduced in size drastically and now contains 30
cells with robust expression information. The new celltype annotation is
stored in the “celltype” column. The column “grouped” contains the
number of cells merged in this metacell while the “percentage” column
stores the abundanceScore of this metacell.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metacells</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 22299 32 </span></span>
<span><span class="co">#&gt; metadata(1): __name__</span></span>
<span><span class="co">#&gt; assays(3): counts scaled round</span></span>
<span><span class="co">#&gt; rownames(22299): RP11-34P13.7 FO538757.3 ... CTD-3222D19.10</span></span>
<span><span class="co">#&gt;   CTC-273B12.7</span></span>
<span><span class="co">#&gt; rowData names(5): excluded_gene clean_gene forbidden_gene</span></span>
<span><span class="co">#&gt;   pre_feature_gene feature_gene</span></span>
<span><span class="co">#&gt; colnames(32): 0 1 ... 31 32</span></span>
<span><span class="co">#&gt; colData names(5): grouped pile candidate celltype percentage</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Constantin Zackl.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
